{
  "cells": [
    {
      "cell_type": "markdown",
      "source": [
        "# PubMed/Biorxiv/Chemrxiv/Medrxiv scraper\n",
        "\n",
        "This script checks several preprint servers as well as PubMed for new papers and determines whether they may be of interest to a reader. The model here uses PubMedBERT embeddings of the article/preprint abstract - specifically, the beginning-of-sentence and end-of-sentence tokens - and feeds them to a logistic regression model that is retrained daily on annotated data. In plain english, I provide a bunch of examples of papers that interest me and papers that don't, and use that to determine whether new papers will or won't also interest me. A downside is that this has lots of false positives, but those don't take long to filter through."
      ],
      "metadata": {
        "id": "fMRr71ZUpxnG"
      }
    },
    {
      "cell_type": "code",
      "execution_count": 1,
      "metadata": {
        "id": "-vbeXtElchqb"
      },
      "outputs": [],
      "source": [
        "%%capture\n",
        "!pip install pymed==0.8.9\n",
        "!pip install paperscraper==0.2.10\n",
        "!git clone https://github.com/delalamo/dda_scripts.git"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 2,
      "metadata": {
        "id": "raKAtXnq0n6g",
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "0d2d7be4-3047-4579-a226-0ee1bb3d3ea6"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "WARNING:paperscraper.load_dumps: No dump found for biorxiv. Skipping entry.\n",
            "WARNING:paperscraper.load_dumps: No dump found for chemrxiv. Skipping entry.\n",
            "WARNING:paperscraper.load_dumps: No dump found for medrxiv. Skipping entry.\n",
            "WARNING:paperscraper.load_dumps: No dumps found for either biorxiv or medrxiv. Consider using paperscraper.get_dumps.* to fetch the dumps.\n"
          ]
        }
      ],
      "source": [
        "import torch\n",
        "import pandas as pd\n",
        "from sklearn.linear_model import LogisticRegression\n",
        "import textwrap\n",
        "from pymed import PubMed\n",
        "from paperscraper.get_dumps import biorxiv, medrxiv, chemrxiv\n",
        "import json\n",
        "import numpy as np\n",
        "from datetime import datetime, timedelta\n",
        "from transformers import AutoTokenizer, AutoModelForMaskedLM"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 3,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 326,
          "referenced_widgets": [
            "a4412ef094074c5d99da2d1b533f8490",
            "60fe81830a6b4caf9b21ec5622f12864",
            "c96b2768610d4b28a76c95fba0fbfa9a",
            "abe277a080104b1983d6fb2c155ac3a3",
            "d0b51d7579934c169ede97fcfa52f43a",
            "a305d73afe94405a8abf721f810cd864",
            "ce5194d96a2942998c6415fcf052a3b3",
            "321ce7a551c74dd191f0dfade58aa7d0",
            "38fa22afb65243228a3542a1ff53ce2d",
            "12068e8a0e6342e2903266b863be0b99",
            "d3d6103e675a4995b9fa6ce3be58d78b",
            "0f33d816e34d4edd9fd5e27369ba538b",
            "8a74492f5db84a18bbe17ebdc216c991",
            "eab69e96b17e4849846e39b4732ccc63",
            "287ae5ff62c242f7991d8c5f60f79d29",
            "a9494b04005a45d39fdb4cd3a98e3f2a",
            "ed12dab7b5a14c7ca67162934d19f44b",
            "8f5cd37e602a4fc2a2b104a8f1913b70",
            "293c56b6e70a462e86e90acec0295201",
            "6516f3039c074d21b7eae37edd97a7d4",
            "b6726896f1894f72b365049ddb5c22b7",
            "484a6db5c1294faa9f3c76e2202d2cc2",
            "a4fb8c79d97a4385944c4530fd90935a",
            "df923834ed854d54aae0655efe2271c4",
            "d83dab517c7241cba4f5423fc7328ad2",
            "c300203ff26849858c7e72a39d1f2316",
            "569a1eebc8394fc0ab53e616df96f155",
            "c9e20a1a2b71498c80b738036f471167",
            "d6a3054e3e194337862c421b33124fd1",
            "4b24874580dd419bb289011f5e9c7ea9",
            "43d348bd3cc44af283f89d92b2e99d6e",
            "5e5387dbf7a04ada9c1508f811d597c9",
            "c94247ead32d4182b03bd7bcbb35241d",
            "a1f9c298cc2c4f7d8db6a014940d5433",
            "66b2bf28fb204af2b8d7c617936b10cb",
            "4ac3ddb33f5841f28b6577256d96e31b",
            "bc5007701a54460b8f85f947cefcb7e8",
            "f781816a03294a63b0942c880e72e0b0",
            "7ff1d81e3b004e9c90d9cc5e5f7f0172",
            "1d294647b2bc4a1685446e7e09910fae",
            "8fe8fd9ae3f242a3951f39074d1fea7b",
            "34971c178aca4603893c5a835d4bace0",
            "0e660b59b87e40b1bcedec32b748f81d",
            "584bd5223e274a2bbb626b5932eb603a"
          ]
        },
        "id": "eUv3VpcAy5tQ",
        "outputId": "ece4c8b6-e522-4449-8929-feec7abe3aa6"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "/usr/local/lib/python3.10/dist-packages/huggingface_hub/utils/_token.py:88: UserWarning: \n",
            "The secret `HF_TOKEN` does not exist in your Colab secrets.\n",
            "To authenticate with the Hugging Face Hub, create a token in your settings tab (https://huggingface.co/settings/tokens), set it as secret in your Google Colab and restart your session.\n",
            "You will be able to reuse this secret in all of your notebooks.\n",
            "Please note that authentication is recommended but still optional to access public models or datasets.\n",
            "  warnings.warn(\n"
          ]
        },
        {
          "output_type": "display_data",
          "data": {
            "text/plain": [
              "tokenizer_config.json:   0%|          | 0.00/28.0 [00:00<?, ?B/s]"
            ],
            "application/vnd.jupyter.widget-view+json": {
              "version_major": 2,
              "version_minor": 0,
              "model_id": "a4412ef094074c5d99da2d1b533f8490"
            }
          },
          "metadata": {}
        },
        {
          "output_type": "display_data",
          "data": {
            "text/plain": [
              "config.json:   0%|          | 0.00/385 [00:00<?, ?B/s]"
            ],
            "application/vnd.jupyter.widget-view+json": {
              "version_major": 2,
              "version_minor": 0,
              "model_id": "0f33d816e34d4edd9fd5e27369ba538b"
            }
          },
          "metadata": {}
        },
        {
          "output_type": "display_data",
          "data": {
            "text/plain": [
              "vocab.txt:   0%|          | 0.00/226k [00:00<?, ?B/s]"
            ],
            "application/vnd.jupyter.widget-view+json": {
              "version_major": 2,
              "version_minor": 0,
              "model_id": "a4fb8c79d97a4385944c4530fd90935a"
            }
          },
          "metadata": {}
        },
        {
          "output_type": "display_data",
          "data": {
            "text/plain": [
              "pytorch_model.bin:   0%|          | 0.00/440M [00:00<?, ?B/s]"
            ],
            "application/vnd.jupyter.widget-view+json": {
              "version_major": 2,
              "version_minor": 0,
              "model_id": "a1f9c298cc2c4f7d8db6a014940d5433"
            }
          },
          "metadata": {}
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "Some weights of the model checkpoint at microsoft/BiomedNLP-BiomedBERT-base-uncased-abstract-fulltext were not used when initializing BertForMaskedLM: ['cls.seq_relationship.bias', 'bert.pooler.dense.bias', 'bert.pooler.dense.weight', 'cls.seq_relationship.weight']\n",
            "- This IS expected if you are initializing BertForMaskedLM from the checkpoint of a model trained on another task or with another architecture (e.g. initializing a BertForSequenceClassification model from a BertForPreTraining model).\n",
            "- This IS NOT expected if you are initializing BertForMaskedLM from the checkpoint of a model that you expect to be exactly identical (initializing a BertForSequenceClassification model from a BertForSequenceClassification model).\n"
          ]
        }
      ],
      "source": [
        "# Download LLM for encoding\n",
        "tokenizer = AutoTokenizer.from_pretrained(\"microsoft/BiomedNLP-BiomedBERT-base-uncased-abstract-fulltext\")\n",
        "model = AutoModelForMaskedLM.from_pretrained(\"microsoft/BiomedNLP-BiomedBERT-base-uncased-abstract-fulltext\").to(\"cuda\")"
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "def process(abstract: str) -> np.ndarray:\n",
        "  \"\"\" Tokenize and encode abstract, return concatenated BOS and EOS array\n",
        "  Token vector must be truncated at 512 (incl BOS & EOS) due to LLM limitations\n",
        "  \"\"\"\n",
        "  inputs = tokenizer(abstract, return_tensors=\"pt\").to(\"cuda\")\n",
        "  inputs = {k: torch.concat((v[:, :511], v[:, -1].unsqueeze(-1)), dim=-1) for k, v in inputs.items()}\n",
        "  hidden_states = model(**inputs, output_hidden_states=True).hidden_states\n",
        "  return torch.concat((hidden_states[-1][0, 0, :], hidden_states[-1][0, -1, :])).detach().cpu().numpy()\n",
        "\n",
        "def format_date(date, sep):\n",
        "  assert len(sep) == 1\n",
        "  return sep.join([date.strftime(f\"%{x}\") for x in \"Ymd\"])\n",
        "\n",
        "def format_dates(dates, sep):\n",
        "  return [format_date(d, sep) for d in dates]"
      ],
      "metadata": {
        "id": "riyqL4S0kqwi"
      },
      "execution_count": 4,
      "outputs": []
    },
    {
      "cell_type": "markdown",
      "source": [
        "I no longer check on weekends, so searches carried out on Monday must also check articles/preprints deposited over the weekend."
      ],
      "metadata": {
        "id": "22Om9qzusHtc"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "yesterday = datetime.now() - timedelta(days = 1)\n",
        "\n",
        "start_rxivs = format_date(yesterday, \"-\")\n",
        "end_rxivs = start_rxivs\n",
        "\n",
        "pubmed_days = [format_date(yesterday, \"/\")]\n",
        "\n",
        "# check if we're catching up on weekend pubs\n",
        "if yesterday.weekday() == 6: # 6 is sunday\n",
        "  saturday = yesterday - timedelta(days = 1)\n",
        "  friday = yesterday - timedelta(days = 2)\n",
        "\n",
        "  start_rxivs = format_date(friday, \"-\")\n",
        "  pubmed_days.extend(format_dates([friday, saturday], \"/\"))"
      ],
      "metadata": {
        "id": "c_9I7Ppje96u"
      },
      "execution_count": 5,
      "outputs": []
    },
    {
      "cell_type": "code",
      "execution_count": 6,
      "metadata": {
        "id": "iyvizsnD0AjN"
      },
      "outputs": [],
      "source": [
        "df = pd.read_csv(\"dda_scripts/extras/annotated_abstracts.tsv\", sep=\"\\t\").dropna(subset=\"label\")"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 7,
      "metadata": {
        "id": "O1IkZvMgy8JW"
      },
      "outputs": [],
      "source": [
        "labels = []\n",
        "reps = []\n",
        "with torch.no_grad():\n",
        "  for i, row in df.iterrows():\n",
        "    reps.append(process(row[\"abstract\"]))\n",
        "    labels.append(row[\"label\"])\n",
        "clf = LogisticRegression(random_state=0, max_iter=10000).fit(reps, labels)"
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "medrxiv(begin_date=start_rxivs, end_date=end_rxivs, save_path=\"medrxiv.jsonl\")\n",
        "biorxiv(begin_date=start_rxivs, end_date=end_rxivs, save_path=\"biorxiv.jsonl\")\n",
        "chemrxiv(begin_date=start_rxivs, end_date=end_rxivs, save_path=\"chemrxiv.jsonl\")"
      ],
      "metadata": {
        "id": "qBtm-qukXzFP",
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "375931f3-e4bf-4a55-c305-7eaec921f826"
      },
      "execution_count": 8,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "51it [00:40,  1.26it/s]\n",
            "184it [02:45,  1.11it/s]\n",
            "26it [00:24,  1.06it/s]\n",
            "100%|██████████| 28/28 [00:00<00:00, 3456.17it/s]\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "data = {\"Title\": [], \"Abstract\": [], \"Probability\": [], \"Journal\": []}"
      ],
      "metadata": {
        "id": "45wd33hkUndl"
      },
      "execution_count": 9,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "for jsonfile in [\"medrxiv.jsonl\", \"biorxiv.jsonl\", \"chemrxiv.jsonl\"]:\n",
        "  with open(jsonfile) as infile:\n",
        "    for line in infile:\n",
        "      l = json.loads(line)\n",
        "      abstract = l[\"abstract\"].replace(\"\\n\", \" \")\n",
        "      reps = process(abstract)\n",
        "      prob = clf.predict_proba(reps.reshape(1, -1))[:, 1].item()\n",
        "      data[\"Title\"].append(l[\"title\"])\n",
        "      data[\"Abstract\"].append(abstract)\n",
        "      data[\"Probability\"].append(prob)\n",
        "      data[\"Journal\"].append(jsonfile.split(\".\")[0])"
      ],
      "metadata": {
        "id": "-nK4-7HpT8TP"
      },
      "execution_count": 10,
      "outputs": []
    },
    {
      "cell_type": "code",
      "execution_count": 11,
      "metadata": {
        "id": "rRVeo_QQPW4e"
      },
      "outputs": [],
      "source": [
        "for date in pubmed_days:\n",
        "  pubmed = PubMed(tool=\"MyTool\", email=\"my@email.address\")\n",
        "  search_query = f\"{date}[PDAT]\"\n",
        "  results = pubmed.query(search_query, max_results=500000)\n",
        "  errors = []\n",
        "  for i, article in enumerate(results):\n",
        "    if article.abstract is None:\n",
        "      continue\n",
        "    reps = process(article.abstract)\n",
        "    prob = clf.predict_proba(reps.reshape(1, -1))[:, 1].item()\n",
        "    data[\"Title\"].append(article.title)\n",
        "    data[\"Abstract\"].append(article.abstract.replace(\"\\n\", \" \"))\n",
        "    data[\"Probability\"].append(prob)\n",
        "    data[\"Journal\"].append(article.journal.strip().replace(\"\\n\", \" \"))"
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "cutoff = 0.05\n",
        "\n",
        "papers = pd.DataFrame.from_dict(data)\n",
        "papers = papers.sort_values(\"Probability\", ascending=False)\n",
        "papers[papers[\"Probability\"] >= cutoff]"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 373
        },
        "id": "V9i_49Mnowrb",
        "outputId": "7ef15de8-35d2-4fef-ddde-069b147103fc"
      },
      "execution_count": 12,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "                                                  Title  \\\n",
              "2077  ProSTAGE: Predicting Effects of Mutations on P...   \n",
              "160   ProtHyena: A fast and efficient foundation pro...   \n",
              "98    The blobulator: a webtool for identification a...   \n",
              "672   Conservation of Hot Spots and Ligand Binding S...   \n",
              "251   Molecular Graph Transformer: Stepping Beyond A...   \n",
              "909   Implicit model to capture electrostatic featur...   \n",
              "2064  Searching for Structure: Characterizing the Pr...   \n",
              "117   Orthogonalized human protease control of secre...   \n",
              "496   MARS an improved de novo peptide candidate sel...   \n",
              "834   Viruses traverse the human proteome through pe...   \n",
              "773      The Site/Group Extended Data format and tools.   \n",
              "1615  Structure of the two-component S-layer of the ...   \n",
              "171   Current limitations in predicting mRNA transla...   \n",
              "78    Biophysical principles predict fitness of SARS...   \n",
              "2057  AMBERff at Scale: Multimillion-Atom Simulation...   \n",
              "161   Pre-trained Maldi Transformers improve MALDI-T...   \n",
              "2065  Active Learning Approach for Guiding Site-of-M...   \n",
              "585   Modular Approach for Rapid Identification of R...   \n",
              "1516    In-Cell Association of a Bioorthogonal Tubulin.   \n",
              "1550  Hexagonal Lattices of HIV Capsid Proteins Expl...   \n",
              "557   Robustness and reproducibility for AI learning...   \n",
              "995    SMEMO: Social Memory for Trajectory Forecasting.   \n",
              "476   Deep learning based CETSA feature prediction c...   \n",
              "817   What to Make of Zero: Resolving the Statistica...   \n",
              "297   Development of Better Aptamers: Structured Lib...   \n",
              "928   Inferred regulons are consistent with regulato...   \n",
              "1513  A benchmark of optimization solvers for genome...   \n",
              "1840  Genetic algorithm-based semisupervised convolu...   \n",
              "101   High-Throughput Identification of Calcium Regu...   \n",
              "238   Accurate Determination of Reactivity Ratios fo...   \n",
              "174   Statistical inference with a manifold-constrai...   \n",
              "371   An error correction strategy for image reconst...   \n",
              "89    Metagenome profiling and containment estimatio...   \n",
              "69    Biophysical characterization and analysis of a...   \n",
              "46    DODGE: Automated point source bacterial outbre...   \n",
              "2034  Docking for Molecules That Bind in a Symmetric...   \n",
              "175   Evaluation of Deep Learning for predicting ric...   \n",
              "370             Language models for quantum simulation.   \n",
              "2027  Aminopyrimidine Derivatives as Multiflavivirus...   \n",
              "12    Fitness models provide accurate short-term for...   \n",
              "1432  Metastable States in the Hinge-Bending Landsca...   \n",
              "235   ChemPrice, a Python package for automated chem...   \n",
              "248   Assessing the Catalytic Role of Native Glucago...   \n",
              "2207         Polarizable AMOEBA Model for Simulating Mg   \n",
              "140   Aggregation of recount3 RNA-seq data improves ...   \n",
              "88    CellTracksColab--A platform for compiling, ana...   \n",
              "54    TrieDedup: A fast trie-based deduplication alg...   \n",
              "91    fastDFE: fast and flexible inference of the di...   \n",
              "2096  pSPICA Force Field Extended for Proteins and P...   \n",
              "169   Cell Type-Agnostic Optical Perturbation Screen...   \n",
              "76    A mean-field theory for predicting single poly...   \n",
              "507                 Segment anything in medical images.   \n",
              "\n",
              "                                               Abstract  Probability  \\\n",
              "2077  Protein thermodynamic stability is essential t...     0.962591   \n",
              "160   The emergence of self-supervised deep language...     0.955628   \n",
              "98    MotivationClusters of hydrophobic residues are...     0.912446   \n",
              "672   The neural network-based program AlphaFold2 (A...     0.910195   \n",
              "251   Graph Neural Networks (GNNs) have revolutioniz...     0.889386   \n",
              "909   Membrane protein structure prediction and desi...     0.883075   \n",
              "2064  The identification and characterization of the...     0.843904   \n",
              "117   Synthetic circuits that regulate protein secre...     0.830171   \n",
              "496   Understanding the nature and extent of non-can...     0.777074   \n",
              "834   We present a drug design strategy based on str...     0.607981   \n",
              "773   Comparative sequence analysis permits unravell...     0.606647   \n",
              "1615  Surface layers (S-layers) are resilient two-di...     0.567400   \n",
              "171   Background: The design of nucleotide sequences...     0.549740   \n",
              "78    SARS-CoV-2 employs its spike protein's recepto...     0.502073   \n",
              "2057  All-atom molecular dynamics (MD) simulations a...     0.477494   \n",
              "161   For the last decade, matrix-assisted laser des...     0.426609   \n",
              "2065  The ability to determine and predict metabolic...     0.398495   \n",
              "585   Detection of metabolites in real time and in w...     0.397931   \n",
              "1516  Studies of proteins from one organism in anoth...     0.363469   \n",
              "1550  HIV capsid proteins (CAs) may self-assemble in...     0.272752   \n",
              "557   Artificial intelligence (AI) techniques are in...     0.244835   \n",
              "995   Effective modeling of human interactions is of...     0.237966   \n",
              "476   Mass spectrometry-coupled cellular thermal shi...     0.230000   \n",
              "817   We introduce the self-relative binding free en...     0.217719   \n",
              "297   Systematic evolution of ligands by exponential...     0.217622   \n",
              "928   The transcriptional regulatory network (TRN) o...     0.177125   \n",
              "1513  Genome-scale metabolic modeling is a powerful ...     0.143057   \n",
              "1840  As a non-destructive sensing technique, Raman ...     0.133222   \n",
              "101   Calcium ions play important roles in nearly ev...     0.127332   \n",
              "238   An understanding of monomer sequence is requir...     0.122595   \n",
              "174   Across a range of biological processes, cells ...     0.120820   \n",
              "371   By pairing adjacent molecules in situ and then...     0.118092   \n",
              "89    Profiling metagenomes against databases allows...     0.116137   \n",
              "69    Chorismate mutases have extensively been used ...     0.095437   \n",
              "46    The reliable and timely recognition of outbrea...     0.092339   \n",
              "2034  Discovering ligands for amyloid fibrils, such ...     0.086520   \n",
              "175   Structural variants (SVs) such as deletions, i...     0.084856   \n",
              "370   A key challenge in the effort to simulate toda...     0.084691   \n",
              "2027  Around three billion people are at risk of inf...     0.082405   \n",
              "12    Genomic surveillance of pathogen evolution is ...     0.080880   \n",
              "1432  Many enzymes undergo major conformational chan...     0.076631   \n",
              "235   The open-access movement in chemistry has led ...     0.072499   \n",
              "248   Glucagon stands out as a pivotal peptide hormo...     0.071396   \n",
              "2207  Molecular mechanics (MM) simulations have the ...     0.066696   \n",
              "140   Background: Gene co-expression networks (GCNs)...     0.066014   \n",
              "88    In life sciences, tracking objects from movies...     0.064519   \n",
              "54    Background: High-throughput sequencing is a po...     0.060852   \n",
              "91    Estimating the distribution of fitness effects...     0.058470   \n",
              "2096  Many coarse-grained (CG) molecular dynamics (M...     0.058184   \n",
              "169   Genome-scale perturbation screening is widely ...     0.055546   \n",
              "76    Macromolecular crowding can induce the collaps...     0.050925   \n",
              "507   Medical image segmentation is a critical compo...     0.050309   \n",
              "\n",
              "                                                Journal  \n",
              "2077       Journal of chemical information and modeling  \n",
              "160                                             biorxiv  \n",
              "98                                              biorxiv  \n",
              "672        Journal of chemical information and modeling  \n",
              "251                                            chemrxiv  \n",
              "909                          PLoS computational biology  \n",
              "2064       Journal of chemical information and modeling  \n",
              "117                                             biorxiv  \n",
              "496                               Nature communications  \n",
              "834   Proceedings of the National Academy of Science...  \n",
              "773                        Genome biology and evolution  \n",
              "1615                                              eLife  \n",
              "171                                             biorxiv  \n",
              "78                                              biorxiv  \n",
              "2057       Journal of chemical information and modeling  \n",
              "161                                             biorxiv  \n",
              "2065       Journal of chemical information and modeling  \n",
              "585                                         ACS sensors  \n",
              "1516                                  Biomacromolecules  \n",
              "1550               The journal of physical chemistry. B  \n",
              "557                                  Scientific reports  \n",
              "995   IEEE transactions on pattern analysis and mach...  \n",
              "476                                  Scientific reports  \n",
              "817          Journal of chemical theory and computation  \n",
              "297    Angewandte Chemie (International ed. in English)  \n",
              "928                          PLoS computational biology  \n",
              "1513                                           mSystems  \n",
              "1840                   Biotechnology and bioengineering  \n",
              "101                                             biorxiv  \n",
              "238                                            chemrxiv  \n",
              "174                                             biorxiv  \n",
              "371                        Nature computational science  \n",
              "89                                              biorxiv  \n",
              "69                                              biorxiv  \n",
              "46                                              medrxiv  \n",
              "2034       Journal of chemical information and modeling  \n",
              "175                                             biorxiv  \n",
              "370                        Nature computational science  \n",
              "2027       Journal of chemical information and modeling  \n",
              "12                                              medrxiv  \n",
              "1432          The journal of physical chemistry letters  \n",
              "235                                            chemrxiv  \n",
              "248                                            chemrxiv  \n",
              "2207       Journal of chemical information and modeling  \n",
              "140                                             biorxiv  \n",
              "88                                              biorxiv  \n",
              "54                                              biorxiv  \n",
              "91                                              biorxiv  \n",
              "2096       Journal of chemical information and modeling  \n",
              "169                                             biorxiv  \n",
              "76                                              biorxiv  \n",
              "507                               Nature communications  "
            ],
            "text/html": [
              "\n",
              "  <div id=\"df-90e831aa-5911-4685-b303-358e79b67ffe\" class=\"colab-df-container\">\n",
              "    <div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>Title</th>\n",
              "      <th>Abstract</th>\n",
              "      <th>Probability</th>\n",
              "      <th>Journal</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>2077</th>\n",
              "      <td>ProSTAGE: Predicting Effects of Mutations on P...</td>\n",
              "      <td>Protein thermodynamic stability is essential t...</td>\n",
              "      <td>0.962591</td>\n",
              "      <td>Journal of chemical information and modeling</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>160</th>\n",
              "      <td>ProtHyena: A fast and efficient foundation pro...</td>\n",
              "      <td>The emergence of self-supervised deep language...</td>\n",
              "      <td>0.955628</td>\n",
              "      <td>biorxiv</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>98</th>\n",
              "      <td>The blobulator: a webtool for identification a...</td>\n",
              "      <td>MotivationClusters of hydrophobic residues are...</td>\n",
              "      <td>0.912446</td>\n",
              "      <td>biorxiv</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>672</th>\n",
              "      <td>Conservation of Hot Spots and Ligand Binding S...</td>\n",
              "      <td>The neural network-based program AlphaFold2 (A...</td>\n",
              "      <td>0.910195</td>\n",
              "      <td>Journal of chemical information and modeling</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>251</th>\n",
              "      <td>Molecular Graph Transformer: Stepping Beyond A...</td>\n",
              "      <td>Graph Neural Networks (GNNs) have revolutioniz...</td>\n",
              "      <td>0.889386</td>\n",
              "      <td>chemrxiv</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>909</th>\n",
              "      <td>Implicit model to capture electrostatic featur...</td>\n",
              "      <td>Membrane protein structure prediction and desi...</td>\n",
              "      <td>0.883075</td>\n",
              "      <td>PLoS computational biology</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>2064</th>\n",
              "      <td>Searching for Structure: Characterizing the Pr...</td>\n",
              "      <td>The identification and characterization of the...</td>\n",
              "      <td>0.843904</td>\n",
              "      <td>Journal of chemical information and modeling</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>117</th>\n",
              "      <td>Orthogonalized human protease control of secre...</td>\n",
              "      <td>Synthetic circuits that regulate protein secre...</td>\n",
              "      <td>0.830171</td>\n",
              "      <td>biorxiv</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>496</th>\n",
              "      <td>MARS an improved de novo peptide candidate sel...</td>\n",
              "      <td>Understanding the nature and extent of non-can...</td>\n",
              "      <td>0.777074</td>\n",
              "      <td>Nature communications</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>834</th>\n",
              "      <td>Viruses traverse the human proteome through pe...</td>\n",
              "      <td>We present a drug design strategy based on str...</td>\n",
              "      <td>0.607981</td>\n",
              "      <td>Proceedings of the National Academy of Science...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>773</th>\n",
              "      <td>The Site/Group Extended Data format and tools.</td>\n",
              "      <td>Comparative sequence analysis permits unravell...</td>\n",
              "      <td>0.606647</td>\n",
              "      <td>Genome biology and evolution</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1615</th>\n",
              "      <td>Structure of the two-component S-layer of the ...</td>\n",
              "      <td>Surface layers (S-layers) are resilient two-di...</td>\n",
              "      <td>0.567400</td>\n",
              "      <td>eLife</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>171</th>\n",
              "      <td>Current limitations in predicting mRNA transla...</td>\n",
              "      <td>Background: The design of nucleotide sequences...</td>\n",
              "      <td>0.549740</td>\n",
              "      <td>biorxiv</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>78</th>\n",
              "      <td>Biophysical principles predict fitness of SARS...</td>\n",
              "      <td>SARS-CoV-2 employs its spike protein's recepto...</td>\n",
              "      <td>0.502073</td>\n",
              "      <td>biorxiv</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>2057</th>\n",
              "      <td>AMBERff at Scale: Multimillion-Atom Simulation...</td>\n",
              "      <td>All-atom molecular dynamics (MD) simulations a...</td>\n",
              "      <td>0.477494</td>\n",
              "      <td>Journal of chemical information and modeling</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>161</th>\n",
              "      <td>Pre-trained Maldi Transformers improve MALDI-T...</td>\n",
              "      <td>For the last decade, matrix-assisted laser des...</td>\n",
              "      <td>0.426609</td>\n",
              "      <td>biorxiv</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>2065</th>\n",
              "      <td>Active Learning Approach for Guiding Site-of-M...</td>\n",
              "      <td>The ability to determine and predict metabolic...</td>\n",
              "      <td>0.398495</td>\n",
              "      <td>Journal of chemical information and modeling</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>585</th>\n",
              "      <td>Modular Approach for Rapid Identification of R...</td>\n",
              "      <td>Detection of metabolites in real time and in w...</td>\n",
              "      <td>0.397931</td>\n",
              "      <td>ACS sensors</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1516</th>\n",
              "      <td>In-Cell Association of a Bioorthogonal Tubulin.</td>\n",
              "      <td>Studies of proteins from one organism in anoth...</td>\n",
              "      <td>0.363469</td>\n",
              "      <td>Biomacromolecules</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1550</th>\n",
              "      <td>Hexagonal Lattices of HIV Capsid Proteins Expl...</td>\n",
              "      <td>HIV capsid proteins (CAs) may self-assemble in...</td>\n",
              "      <td>0.272752</td>\n",
              "      <td>The journal of physical chemistry. B</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>557</th>\n",
              "      <td>Robustness and reproducibility for AI learning...</td>\n",
              "      <td>Artificial intelligence (AI) techniques are in...</td>\n",
              "      <td>0.244835</td>\n",
              "      <td>Scientific reports</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>995</th>\n",
              "      <td>SMEMO: Social Memory for Trajectory Forecasting.</td>\n",
              "      <td>Effective modeling of human interactions is of...</td>\n",
              "      <td>0.237966</td>\n",
              "      <td>IEEE transactions on pattern analysis and mach...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>476</th>\n",
              "      <td>Deep learning based CETSA feature prediction c...</td>\n",
              "      <td>Mass spectrometry-coupled cellular thermal shi...</td>\n",
              "      <td>0.230000</td>\n",
              "      <td>Scientific reports</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>817</th>\n",
              "      <td>What to Make of Zero: Resolving the Statistica...</td>\n",
              "      <td>We introduce the self-relative binding free en...</td>\n",
              "      <td>0.217719</td>\n",
              "      <td>Journal of chemical theory and computation</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>297</th>\n",
              "      <td>Development of Better Aptamers: Structured Lib...</td>\n",
              "      <td>Systematic evolution of ligands by exponential...</td>\n",
              "      <td>0.217622</td>\n",
              "      <td>Angewandte Chemie (International ed. in English)</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>928</th>\n",
              "      <td>Inferred regulons are consistent with regulato...</td>\n",
              "      <td>The transcriptional regulatory network (TRN) o...</td>\n",
              "      <td>0.177125</td>\n",
              "      <td>PLoS computational biology</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1513</th>\n",
              "      <td>A benchmark of optimization solvers for genome...</td>\n",
              "      <td>Genome-scale metabolic modeling is a powerful ...</td>\n",
              "      <td>0.143057</td>\n",
              "      <td>mSystems</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1840</th>\n",
              "      <td>Genetic algorithm-based semisupervised convolu...</td>\n",
              "      <td>As a non-destructive sensing technique, Raman ...</td>\n",
              "      <td>0.133222</td>\n",
              "      <td>Biotechnology and bioengineering</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>101</th>\n",
              "      <td>High-Throughput Identification of Calcium Regu...</td>\n",
              "      <td>Calcium ions play important roles in nearly ev...</td>\n",
              "      <td>0.127332</td>\n",
              "      <td>biorxiv</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>238</th>\n",
              "      <td>Accurate Determination of Reactivity Ratios fo...</td>\n",
              "      <td>An understanding of monomer sequence is requir...</td>\n",
              "      <td>0.122595</td>\n",
              "      <td>chemrxiv</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>174</th>\n",
              "      <td>Statistical inference with a manifold-constrai...</td>\n",
              "      <td>Across a range of biological processes, cells ...</td>\n",
              "      <td>0.120820</td>\n",
              "      <td>biorxiv</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>371</th>\n",
              "      <td>An error correction strategy for image reconst...</td>\n",
              "      <td>By pairing adjacent molecules in situ and then...</td>\n",
              "      <td>0.118092</td>\n",
              "      <td>Nature computational science</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>89</th>\n",
              "      <td>Metagenome profiling and containment estimatio...</td>\n",
              "      <td>Profiling metagenomes against databases allows...</td>\n",
              "      <td>0.116137</td>\n",
              "      <td>biorxiv</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>69</th>\n",
              "      <td>Biophysical characterization and analysis of a...</td>\n",
              "      <td>Chorismate mutases have extensively been used ...</td>\n",
              "      <td>0.095437</td>\n",
              "      <td>biorxiv</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>46</th>\n",
              "      <td>DODGE: Automated point source bacterial outbre...</td>\n",
              "      <td>The reliable and timely recognition of outbrea...</td>\n",
              "      <td>0.092339</td>\n",
              "      <td>medrxiv</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>2034</th>\n",
              "      <td>Docking for Molecules That Bind in a Symmetric...</td>\n",
              "      <td>Discovering ligands for amyloid fibrils, such ...</td>\n",
              "      <td>0.086520</td>\n",
              "      <td>Journal of chemical information and modeling</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>175</th>\n",
              "      <td>Evaluation of Deep Learning for predicting ric...</td>\n",
              "      <td>Structural variants (SVs) such as deletions, i...</td>\n",
              "      <td>0.084856</td>\n",
              "      <td>biorxiv</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>370</th>\n",
              "      <td>Language models for quantum simulation.</td>\n",
              "      <td>A key challenge in the effort to simulate toda...</td>\n",
              "      <td>0.084691</td>\n",
              "      <td>Nature computational science</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>2027</th>\n",
              "      <td>Aminopyrimidine Derivatives as Multiflavivirus...</td>\n",
              "      <td>Around three billion people are at risk of inf...</td>\n",
              "      <td>0.082405</td>\n",
              "      <td>Journal of chemical information and modeling</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>12</th>\n",
              "      <td>Fitness models provide accurate short-term for...</td>\n",
              "      <td>Genomic surveillance of pathogen evolution is ...</td>\n",
              "      <td>0.080880</td>\n",
              "      <td>medrxiv</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1432</th>\n",
              "      <td>Metastable States in the Hinge-Bending Landsca...</td>\n",
              "      <td>Many enzymes undergo major conformational chan...</td>\n",
              "      <td>0.076631</td>\n",
              "      <td>The journal of physical chemistry letters</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>235</th>\n",
              "      <td>ChemPrice, a Python package for automated chem...</td>\n",
              "      <td>The open-access movement in chemistry has led ...</td>\n",
              "      <td>0.072499</td>\n",
              "      <td>chemrxiv</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>248</th>\n",
              "      <td>Assessing the Catalytic Role of Native Glucago...</td>\n",
              "      <td>Glucagon stands out as a pivotal peptide hormo...</td>\n",
              "      <td>0.071396</td>\n",
              "      <td>chemrxiv</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>2207</th>\n",
              "      <td>Polarizable AMOEBA Model for Simulating Mg</td>\n",
              "      <td>Molecular mechanics (MM) simulations have the ...</td>\n",
              "      <td>0.066696</td>\n",
              "      <td>Journal of chemical information and modeling</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>140</th>\n",
              "      <td>Aggregation of recount3 RNA-seq data improves ...</td>\n",
              "      <td>Background: Gene co-expression networks (GCNs)...</td>\n",
              "      <td>0.066014</td>\n",
              "      <td>biorxiv</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>88</th>\n",
              "      <td>CellTracksColab--A platform for compiling, ana...</td>\n",
              "      <td>In life sciences, tracking objects from movies...</td>\n",
              "      <td>0.064519</td>\n",
              "      <td>biorxiv</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>54</th>\n",
              "      <td>TrieDedup: A fast trie-based deduplication alg...</td>\n",
              "      <td>Background: High-throughput sequencing is a po...</td>\n",
              "      <td>0.060852</td>\n",
              "      <td>biorxiv</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>91</th>\n",
              "      <td>fastDFE: fast and flexible inference of the di...</td>\n",
              "      <td>Estimating the distribution of fitness effects...</td>\n",
              "      <td>0.058470</td>\n",
              "      <td>biorxiv</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>2096</th>\n",
              "      <td>pSPICA Force Field Extended for Proteins and P...</td>\n",
              "      <td>Many coarse-grained (CG) molecular dynamics (M...</td>\n",
              "      <td>0.058184</td>\n",
              "      <td>Journal of chemical information and modeling</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>169</th>\n",
              "      <td>Cell Type-Agnostic Optical Perturbation Screen...</td>\n",
              "      <td>Genome-scale perturbation screening is widely ...</td>\n",
              "      <td>0.055546</td>\n",
              "      <td>biorxiv</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>76</th>\n",
              "      <td>A mean-field theory for predicting single poly...</td>\n",
              "      <td>Macromolecular crowding can induce the collaps...</td>\n",
              "      <td>0.050925</td>\n",
              "      <td>biorxiv</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>507</th>\n",
              "      <td>Segment anything in medical images.</td>\n",
              "      <td>Medical image segmentation is a critical compo...</td>\n",
              "      <td>0.050309</td>\n",
              "      <td>Nature communications</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "</div>\n",
              "    <div class=\"colab-df-buttons\">\n",
              "\n",
              "  <div class=\"colab-df-container\">\n",
              "    <button class=\"colab-df-convert\" onclick=\"convertToInteractive('df-90e831aa-5911-4685-b303-358e79b67ffe')\"\n",
              "            title=\"Convert this dataframe to an interactive table.\"\n",
              "            style=\"display:none;\">\n",
              "\n",
              "  <svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\" viewBox=\"0 -960 960 960\">\n",
              "    <path d=\"M120-120v-720h720v720H120Zm60-500h600v-160H180v160Zm220 220h160v-160H400v160Zm0 220h160v-160H400v160ZM180-400h160v-160H180v160Zm440 0h160v-160H620v160ZM180-180h160v-160H180v160Zm440 0h160v-160H620v160Z\"/>\n",
              "  </svg>\n",
              "    </button>\n",
              "\n",
              "  <style>\n",
              "    .colab-df-container {\n",
              "      display:flex;\n",
              "      gap: 12px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert {\n",
              "      background-color: #E8F0FE;\n",
              "      border: none;\n",
              "      border-radius: 50%;\n",
              "      cursor: pointer;\n",
              "      display: none;\n",
              "      fill: #1967D2;\n",
              "      height: 32px;\n",
              "      padding: 0 0 0 0;\n",
              "      width: 32px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert:hover {\n",
              "      background-color: #E2EBFA;\n",
              "      box-shadow: 0px 1px 2px rgba(60, 64, 67, 0.3), 0px 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "      fill: #174EA6;\n",
              "    }\n",
              "\n",
              "    .colab-df-buttons div {\n",
              "      margin-bottom: 4px;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert {\n",
              "      background-color: #3B4455;\n",
              "      fill: #D2E3FC;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert:hover {\n",
              "      background-color: #434B5C;\n",
              "      box-shadow: 0px 1px 3px 1px rgba(0, 0, 0, 0.15);\n",
              "      filter: drop-shadow(0px 1px 2px rgba(0, 0, 0, 0.3));\n",
              "      fill: #FFFFFF;\n",
              "    }\n",
              "  </style>\n",
              "\n",
              "    <script>\n",
              "      const buttonEl =\n",
              "        document.querySelector('#df-90e831aa-5911-4685-b303-358e79b67ffe button.colab-df-convert');\n",
              "      buttonEl.style.display =\n",
              "        google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "\n",
              "      async function convertToInteractive(key) {\n",
              "        const element = document.querySelector('#df-90e831aa-5911-4685-b303-358e79b67ffe');\n",
              "        const dataTable =\n",
              "          await google.colab.kernel.invokeFunction('convertToInteractive',\n",
              "                                                    [key], {});\n",
              "        if (!dataTable) return;\n",
              "\n",
              "        const docLinkHtml = 'Like what you see? Visit the ' +\n",
              "          '<a target=\"_blank\" href=https://colab.research.google.com/notebooks/data_table.ipynb>data table notebook</a>'\n",
              "          + ' to learn more about interactive tables.';\n",
              "        element.innerHTML = '';\n",
              "        dataTable['output_type'] = 'display_data';\n",
              "        await google.colab.output.renderOutput(dataTable, element);\n",
              "        const docLink = document.createElement('div');\n",
              "        docLink.innerHTML = docLinkHtml;\n",
              "        element.appendChild(docLink);\n",
              "      }\n",
              "    </script>\n",
              "  </div>\n",
              "\n",
              "\n",
              "<div id=\"df-014d0b23-29da-4084-9872-facf49cf94e9\">\n",
              "  <button class=\"colab-df-quickchart\" onclick=\"quickchart('df-014d0b23-29da-4084-9872-facf49cf94e9')\"\n",
              "            title=\"Suggest charts\"\n",
              "            style=\"display:none;\">\n",
              "\n",
              "<svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\"viewBox=\"0 0 24 24\"\n",
              "     width=\"24px\">\n",
              "    <g>\n",
              "        <path d=\"M19 3H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zM9 17H7v-7h2v7zm4 0h-2V7h2v10zm4 0h-2v-4h2v4z\"/>\n",
              "    </g>\n",
              "</svg>\n",
              "  </button>\n",
              "\n",
              "<style>\n",
              "  .colab-df-quickchart {\n",
              "      --bg-color: #E8F0FE;\n",
              "      --fill-color: #1967D2;\n",
              "      --hover-bg-color: #E2EBFA;\n",
              "      --hover-fill-color: #174EA6;\n",
              "      --disabled-fill-color: #AAA;\n",
              "      --disabled-bg-color: #DDD;\n",
              "  }\n",
              "\n",
              "  [theme=dark] .colab-df-quickchart {\n",
              "      --bg-color: #3B4455;\n",
              "      --fill-color: #D2E3FC;\n",
              "      --hover-bg-color: #434B5C;\n",
              "      --hover-fill-color: #FFFFFF;\n",
              "      --disabled-bg-color: #3B4455;\n",
              "      --disabled-fill-color: #666;\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart {\n",
              "    background-color: var(--bg-color);\n",
              "    border: none;\n",
              "    border-radius: 50%;\n",
              "    cursor: pointer;\n",
              "    display: none;\n",
              "    fill: var(--fill-color);\n",
              "    height: 32px;\n",
              "    padding: 0;\n",
              "    width: 32px;\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart:hover {\n",
              "    background-color: var(--hover-bg-color);\n",
              "    box-shadow: 0 1px 2px rgba(60, 64, 67, 0.3), 0 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "    fill: var(--button-hover-fill-color);\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart-complete:disabled,\n",
              "  .colab-df-quickchart-complete:disabled:hover {\n",
              "    background-color: var(--disabled-bg-color);\n",
              "    fill: var(--disabled-fill-color);\n",
              "    box-shadow: none;\n",
              "  }\n",
              "\n",
              "  .colab-df-spinner {\n",
              "    border: 2px solid var(--fill-color);\n",
              "    border-color: transparent;\n",
              "    border-bottom-color: var(--fill-color);\n",
              "    animation:\n",
              "      spin 1s steps(1) infinite;\n",
              "  }\n",
              "\n",
              "  @keyframes spin {\n",
              "    0% {\n",
              "      border-color: transparent;\n",
              "      border-bottom-color: var(--fill-color);\n",
              "      border-left-color: var(--fill-color);\n",
              "    }\n",
              "    20% {\n",
              "      border-color: transparent;\n",
              "      border-left-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "    }\n",
              "    30% {\n",
              "      border-color: transparent;\n",
              "      border-left-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "      border-right-color: var(--fill-color);\n",
              "    }\n",
              "    40% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "    }\n",
              "    60% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "    }\n",
              "    80% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "      border-bottom-color: var(--fill-color);\n",
              "    }\n",
              "    90% {\n",
              "      border-color: transparent;\n",
              "      border-bottom-color: var(--fill-color);\n",
              "    }\n",
              "  }\n",
              "</style>\n",
              "\n",
              "  <script>\n",
              "    async function quickchart(key) {\n",
              "      const quickchartButtonEl =\n",
              "        document.querySelector('#' + key + ' button');\n",
              "      quickchartButtonEl.disabled = true;  // To prevent multiple clicks.\n",
              "      quickchartButtonEl.classList.add('colab-df-spinner');\n",
              "      try {\n",
              "        const charts = await google.colab.kernel.invokeFunction(\n",
              "            'suggestCharts', [key], {});\n",
              "      } catch (error) {\n",
              "        console.error('Error during call to suggestCharts:', error);\n",
              "      }\n",
              "      quickchartButtonEl.classList.remove('colab-df-spinner');\n",
              "      quickchartButtonEl.classList.add('colab-df-quickchart-complete');\n",
              "    }\n",
              "    (() => {\n",
              "      let quickchartButtonEl =\n",
              "        document.querySelector('#df-014d0b23-29da-4084-9872-facf49cf94e9 button');\n",
              "      quickchartButtonEl.style.display =\n",
              "        google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "    })();\n",
              "  </script>\n",
              "</div>\n",
              "    </div>\n",
              "  </div>\n"
            ]
          },
          "metadata": {},
          "execution_count": 12
        }
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 13,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "G2gboZUlRuwt",
        "outputId": "33e9323b-93e4-4839-8a03-40dcef7dad79"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "0.9626\tProSTAGE: Predicting Effects of Mutations on Protein Stability by Using Protein Embeddings and Graph Convolutional Networks. (Journal of chemical information and modeling)\tProtein thermodynamic stability is essential to clarify the relationships among structure, function, and interaction. Therefore, developing a faster and more accurate method to predict the impact of the mutations on protein stability is helpful for protein design and understanding the phenotypic variation. Recent studies have shown that protein embedding will be particularly powerful at modeling sequence information with context dependence, such as subcellular localization, variant effect, and secondary structure prediction. Herein, we introduce a novel method, ProSTAGE, which is a deep learning method that fuses structure and sequence embedding to predict protein stability changes upon single point mutations. Our model combines graph-based techniques and language models to predict stability changes. Moreover, ProSTAGE is trained on a larger data set, which is almost twice as large as the most used S2648 data set. It consistently outperforms all existing state-of-the-art methods on mutation-affected problems as benchmarked on several independent data sets. The protein embedding as the prediction input achieves better results than the previous results, which shows the potential of protein language models in predicting the effect of mutations on proteins. ProSTAGE is implemented as a user-friendly web server.\t0\n",
            "0.9556\tProtHyena: A fast and efficient foundation protein language model at single amino acid resolution (biorxiv)\tThe emergence of self-supervised deep language models has revolutionized natural language processing tasks and has recently extended its applications to biological sequence analysis. Traditional models, primarily based on the Transformer and BERT architectures, demonstrate substantial effectiveness in various applications. However, these models are inherently constrained by the attention mechanism's quadratic computational complexity O (L2), limiting their efficiency and the length of context they can process. Addressing these limitations, we introduce ProtHyena, a novel approach that leverages the Hyena operator. This innovative methodology circumvents the constraints imposed by attention mechanisms, thereby reducing the time complexity to a subquadratic, enabling the modeling of extra-long protein sequences at the single amino acid level without the need to compress data. ProtHyena is able to achieve, and in many cases exceed, state-of-the-art results in various downstream tasks with only 10% of the parameters typically required by attention-based models. The architecture of ProtHyena presents a highly efficient solution for training protein predictors, offering a promising avenue for fast and efficient analysis of biological sequences.\t0\n",
            "0.9124\tThe blobulator: a webtool for identification and visual exploration of hydrophobic modularity in protein sequences (biorxiv)\tMotivationClusters of hydrophobic residues are known to promote structured protein stability and drive protein aggregation. Recent work has shown that identifying contiguous hydrophobic residue clusters (termed \"blobs\") has proven useful in both intrinsically disordered protein (IDP) simulation and human genome studies. However, a graphical interface was unavailable.  ResultsHere, we present the blobulator: an interactive and intuitive web interface to detect intrinsic modularity in any protein sequence based on hydrophobicity. We demonstrate three use cases of the blobulator and show how identifying blobs with biologically relevant parameters provides useful information about a globular protein, two orthologous membrane proteins, and an IDP. Other potential applications are discussed, including: predicting protein segments with critical roles in tertiary interactions, providing a definition of local order and disorder with clear edges, and aiding in predicting protein features from sequence.  AvailabilityThe blobulator GUI can be found at www.blobulator.branniganlab.org, and the source code with pip installable command line tool can be found on GitHub at www.GitHub.com/BranniganLab/blobulator.\t0\n",
            "0.9102\tConservation of Hot Spots and Ligand Binding Sites in Protein Models by AlphaFold2. (Journal of chemical information and modeling)\tThe neural network-based program AlphaFold2 (AF2) provides high accuracy structure prediction for a large fraction of globular proteins. An important question is whether these models are accurate enough for reliably docking small ligands. Several recent papers and the results of CASP15 reveal that local conformational errors reduce the success rates of direct ligand docking. Here, we focus on the ability of the models to conserve the location of binding hot spots, regions on the protein surface that significantly contribute to the binding free energy of the protein-ligand interaction. Clusters of hot spots predict the location and even the druggability of binding sites, and hence are important for computational drug discovery. The hot spots are determined by protein mapping that is based on the distribution of small fragment-sized probes on the protein surface and is less sensitive to local conformation than docking. Mapping models taken from the AlphaFold Protein Structure Database show that identifying binding sites is more reliable than docking, but the success rates are still 5% to 10% lower than based on mapping X-ray structures. The drop in accuracy is particularly large for models of multidomain proteins. However, both the model binding sites and the mapping results can be substantially improved by generating AF2 models for the ligand binding domains of interest rather than the entire proteins and even more if using forced sampling with multiple initial seeds. The mapping of such models tends to reach the accuracy of results obtained by mapping the X-ray structures.\t0\n",
            "0.8894\tMolecular Graph Transformer: Stepping Beyond ALIGNN Into Long-Range Interactions (chemrxiv)\tGraph Neural Networks (GNNs) have revolutionized material property prediction by learning directly from the structural information of molecules and materials. However, conventional GNN models rely solely on local atomic interactions, such as bond lengths and angles, neglecting crucial long\u0002range electrostatic forces that a\u001bect certain properties. To address this, we introduce the Molecular Graph Transformer (MGT), a novel GNN architecture that combines local attention mechanisms with message passing on both bond graphs and their line graphs, explicitly capturing long-range interactions. Benchmarking on MatBench and Quantum MOF (QMOF) datasets demonstrates that MGT's improved understanding of electrostatic interactions signi\u001ccantly enhances the prediction accuracy of properties like exfoliation energy and refractive index, while maintaining state-of-the\u0002art performance on all other properties. This breakthrough paves the way for the development of highly accurate and efficient materials design tools across diverse applications. Code is available at: https://github.com/MolecularGraphTransformer/MGT\t0\n",
            "0.8831\tImplicit model to capture electrostatic features of membrane environment. (PLoS computational biology)\tMembrane protein structure prediction and design are challenging due to the complexity of capturing the interactions in the lipid layer, such as those arising from electrostatics. Accurately capturing electrostatic energies in the low-dielectric membrane often requires expensive Poisson-Boltzmann calculations that are not scalable for membrane protein structure prediction and design. In this work, we have developed a fast-to-compute implicit energy function that considers the realistic characteristics of different lipid bilayers, making design calculations tractable. This method captures the impact of the lipid head group using a mean-field-based approach and uses a depth-dependent dielectric constant to characterize the membrane environment. This energy function Franklin2023 (F23) is built upon Franklin2019 (F19), which is based on experimentally derived hydrophobicity scales in the membrane bilayer. We evaluated the performance of F23 on five different tests probing (1) protein orientation in the bilayer, (2) stability, and (3) sequence recovery. Relative to F19, F23 has improved the calculation of the tilt angle of membrane proteins for 90% of WALP peptides, 15% of TM-peptides, and 25% of the adsorbed peptides. The performances for stability and design tests were equivalent for F19 and F23. The speed and calibration of the implicit model will help F23 access biophysical phenomena at long time and length scales and accelerate the membrane protein design pipeline.\t0\n",
            "0.8439\tSearching for Structure: Characterizing the Protein Conformational Landscape with Clustering-Based Algorithms. (Journal of chemical information and modeling)\tThe identification and characterization of the main conformations from a protein population are a challenging and inherently high-dimensional problem. Here, we evaluate the performance of the Secondary sTructural Ensembles with machine LeArning (StELa) double-clustering method, which clusters protein structures based on the relationship between the φ and ψ dihedral angles in a protein backbone and the secondary structure of the protein, thus focusing on the local properties of protein structures. The classification of states as vectors composed of the clusters' indices arising naturally from the Ramachandran plot is followed by the hierarchical clustering of the vectors to allow for the identification of the main features of the corresponding free energy landscape (FEL). We compare the performance of StELa with the established root-mean-squared-deviation (RMSD)-based clustering algorithm, which focuses on global properties of protein structures and with Combinatorial Averaged Transient Structure (CATS), the combinatorial averaged transient structure clustering method based on distributions of the φ and ψ dihedral angle coordinates. Using ensembles of conformations from molecular dynamics simulations of intrinsically disordered proteins (IDPs) of various lengths (tau protein fragments) or short fragments from a globular protein, we show that StELa is the clustering method that identifies many of the minima and relevant energy states around the minima from the corresponding FELs. In contrast, the RMSD-based algorithm yields a large number of clusters that usually cover most of the FEL, thus being unable to distinguish between states, while CATS does not sample well the FELs for long IDPs and fragments from globular proteins.\t0\n",
            "0.8302\tOrthogonalized human protease control of secreted signals (biorxiv)\tSynthetic circuits that regulate protein secretion in human cells could support cell-based therapies by enabling control over local environments. While protein-level circuits enable such potential clinical applications, featuring orthogonality and compactness, their non-human origin poses a potential immunogenic risk. Here, we developed Humanized Drug Induced Regulation of Engineered CyTokines (hDIRECT) as a platform to control cytokine activity exclusively using human-derived proteins. We sourced a specific human protease and its FDA-approved inhibitor. We engineered cytokines whose activities can be activated and abrogated by proteolytic cleavage. We utilized species specificity and re-localization strategies to orthogonalize the cytokines and protease from the human context that they would be deployed in. hDIRECT should enable local cytokine activation to support a variety of cell-based therapies such as muscle regeneration and cancer immunotherapy. Our work offers a proof of concept for the emerging appreciation of humanization in synthetic biology for human health.\t0\n",
            "0.7771\tMARS an improved de novo peptide candidate selection method for non-canonical antigen target discovery in cancer. (Nature communications)\tUnderstanding the nature and extent of non-canonical human leukocyte antigen (HLA) presentation in tumour cells is a priority for target antigen discovery for the development of next generation immunotherapies in cancer. We here employ a de novo mass spectrometric sequencing approach with a refined, MHC-centric analysis strategy to detect non-canonical MHC-associated peptides specific to cancer without any prior knowledge of the target sequence from genomic or RNA sequencing data. Our strategy integrates MHC binding rank, Average local confidence scores, and peptide Retention time prediction for improved de novo candidate Selection; culminating in the machine learning model MARS. We benchmark our model on a large synthetic peptide library dataset and reanalysis of a published dataset of high-quality non-canonical MHC-associated peptide identifications in human cancer. We achieve almost 2-fold improvement for high quality spectral assignments in comparison to de novo sequencing alone with an estimated accuracy of above 85.7% when integrated with a stepwise peptide sequence mapping strategy. Finally, we utilize MARS to detect and validate lncRNA-derived peptides in human cervical tumour resections, demonstrating its suitability to discover novel, immunogenic, non-canonical peptide sequences in primary tumour tissue.\t0\n",
            "0.6080\tViruses traverse the human proteome through peptide interfaces that can be biomimetically leveraged for drug discovery. (Proceedings of the National Academy of Sciences of the United States of America)\tWe present a drug design strategy based on structural knowledge of protein-protein interfaces selected through virus-host coevolution and translated into highly potential small molecules. This approach is grounded on Vinland, the most comprehensive atlas of virus-human protein-protein interactions with annotation of interacting domains. From this inspiration, we identified small viral protein domains responsible for interaction with human proteins. These peptides form a library of new chemical entities used to screen for replication modulators of several pathogens. As a proof of concept, a peptide from a KSHV protein, identified as an inhibitor of influenza virus replication, was translated into a small molecule series with low nanomolar antiviral activity. By targeting the NEET proteins, these molecules turn out to be of therapeutic interest in a nonalcoholic steatohepatitis mouse model with kidney lesions. This study provides a biomimetic framework to design original chemistries targeting cellular proteins, with indications going far beyond infectious diseases.\t0\n",
            "0.6066\tThe Site/Group Extended Data format and tools. (Genome biology and evolution)\tComparative sequence analysis permits unravelling the molecular processes underlying gene evolution. Many statistical methods generate candidate positions within genes, such as fast or slowly-evolving sites, coevolving groups of residues, sites undergoing positive selection or changes in evolutionary rates. Understanding the functional causes of these evolutionary patterns requires combining the results of these analyses and mapping them onto molecular structures, a complex task involving distinct coordinate referential systems. To ease this task, we introduce the site/group extended data (SGED) format, a simple text format to store (groups of) site annotations. We developed a toolset, the SgedTools, which permits SGED file manipulation, creating them from various software outputs and translating coordinates between individual sequences, alignments, and three-dimensional structures. The package also includes a Monte-Carlo procedure to generate random site samples, possibly conditioning on site-specific features. This eases the statistical testing of evolutionary hypotheses, accounting for the structural properties of the encoded molecules.\t0\n",
            "0.5674\tStructure of the two-component S-layer of the archaeon  (eLife)\tSurface layers (S-layers) are resilient two-dimensional protein lattices that encapsulate many bacteria and most archaea. In archaea, S-layers usually form the only structural component of the cell wall and thus act as the final frontier between the cell and its environment. Therefore, S-layers are crucial for supporting microbial life. Notwithstanding their importance, little is known about archaeal S-layers at the atomic level. Here, we combined single particle cryo electron microscopy (cryoEM), cryo electron tomography (cryoET) and Alphafold2 predictions to generate an atomic model of the two-component S-layer of \t0\n",
            "0.5497\tCurrent limitations in predicting mRNA translation with deep learning models (biorxiv)\tBackground: The design of nucleotide sequences with defined properties is long-standing problem in bioengineering. An important application is protein expression, be it in the context of research or the production of mRNA vaccines. The rate of protein synthesis depends on the 5, untranslated region (5,UTR) of the mRNAs, and recently, deep learning models were proposed to predict the translation output of mRNAs from the 5,UTR sequence. At the same time, large data sets of endogenous and reporter mRNA translation have become available. Results: In this study we use complementary data obtained in two different cell types to assess the accuracy and generality of currently available models of translation. We find that while performing well on the data sets on which they were trained, deep learning models do not generalize well to other data sets, in particular of endogenous mRNAs, which differ in many properties from reporter constructs. Conclusions: These differences limit the ability of deep learning models to uncover mechanisms of translation control and to predict the impact of genetic variation. We suggest directions that combine high-throughput measurements and machine learning to unravel mechanisms of translation control and improve construct design.\t0\n",
            "0.5021\tBiophysical principles predict fitness of SARS-CoV-2 variants (biorxiv)\tSARS-CoV-2 employs its spike protein's receptor binding domain (RBD) to enter host cells. The RBD is constantly subjected to immune responses, while requiring efficient binding to host cell receptors for successful infection. However, our understanding of how RBD's biophysical properties contribute to SARS-CoV-2's epidemiological fitness remains largely incomplete. Through a comprehensive approach, comprising large-scale sequence analysis of SARS-CoV-2 variants and the discovery of a fitness function based on binding thermodynamics, we unravel the relationship between the biophysical properties of RBD variants and their contribution to viral fitness. We developed a biophysical model that uses statistical mechanics to map the molecular phenotype space, characterized by binding constants of RBD to ACE2, LY-CoV016, LY-CoV555, REGN10987, and S309, onto a epistatic fitness landscape. We validate our findings through experimentally measured and machine learning (ML) estimated binding affinities, coupled with infectivity data derived from population-level sequencing. Our analysis reveals that this model effectively predicts the fitness of novel RBD variants and can account for the epistatic interactions among mutations, including explaining the later reversal of Q493R. Our study sheds light on the impact of specific mutations on viral fitness and delivers a tool for predicting the future epidemiological trajectory of previously unseen or emerging low frequency variants. These insights offer not only greater understanding of viral evolution but also potentially aid in guiding public health decisions in the battle against COVID-19 and future pandemics.\t0\n",
            "0.4775\tAMBERff at Scale: Multimillion-Atom Simulations with AMBER Force Fields in NAMD. (Journal of chemical information and modeling)\tAll-atom molecular dynamics (MD) simulations are an essential structural biology technique with increasing application to multimillion-atom systems, including viruses and cellular machinery. Classical MD simulations rely on parameter sets, such as the AMBER family of force fields (AMBERff), to accurately describe molecular motion. Here, we present an implementation of AMBERff for use in NAMD that overcomes previous limitations to enable high-performance, massively parallel simulations encompassing up to two billion atoms. Single-point potential energy comparisons and case studies on model systems demonstrate that the implementation produces results that are as accurate as running AMBERff in its native engine.\t0\n",
            "0.4266\tPre-trained Maldi Transformers improve MALDI-TOF MS-based prediction (biorxiv)\tFor the last decade, matrix-assisted laser desportion/ionization time-of-flight mass spectrometry (MALDI-TOF MS) has been the reference method for species identification in clinical microbiology. Hampered by a historical lack of open data, machine learning research towards models specifically adapted to MALDI-TOF MS remains in its infancy. Given the growing complexity of available datasets (such as large-scale antimicrobial resistance prediction), a need for models that (1) are specifically designed for MALDI-TOF MS data, and (2) have high representational capacity, presents itself. Here, we introduce Maldi Transformer, an adaptation of the state-of-the-art transformer architecture to the MALDI-TOF mass spectral domain. We propose the first self-supervised pre-training technique adapted to mass spectra. The technique is based on shuffling peaks across spectra, and pre-training the transformer as a peak discriminator. Extensive benchmarks confirm the efficacy of this novel design. The final result is a model exhibiting state-of-the-art (or competitive) performance on downstream prediction tasks. In addition, we show that Maldi Transformer's identification of noisy spectra may be leveraged towards higher predictive performance. All code supporting this study is distributed on PyPI and is packaged under: https://github.com/gdewael/maldi-nn\t0\n",
            "0.3985\tActive Learning Approach for Guiding Site-of-Metabolism Measurement and Annotation. (Journal of chemical information and modeling)\tThe ability to determine and predict metabolically labile atom positions in a molecule (also called \"sites of metabolism\" or \"SoMs\") is of high interest to the design and optimization of bioactive compounds, such as drugs, agrochemicals, and cosmetics. In recent years, several in silico models for SoM prediction have become available, many of which include a machine-learning component. The bottleneck in advancing these approaches is the coverage of distinct atom environments and rare and complex biotransformation events with high-quality experimental data. Pharmaceutical companies typically have measured metabolism data available for several hundred to several thousand compounds. However, even for metabolism experts, interpreting these data and assigning SoMs are challenging and time-consuming. Therefore, a significant proportion of the potential of the existing metabolism data, particularly in machine learning, remains dormant. Here, we report on the development and validation of an active learning approach that identifies the most informative atoms across molecular data sets for SoM annotation. The active learning approach, built on a highly efficient reimplementation of SoM predictor FAME 3, enables experts to prioritize their SoM experimental measurements and annotation efforts on the most rewarding atom environments. We show that this active learning approach yields competitive SoM predictors while requiring the annotation of only 20% of the atom positions required by FAME 3. The source code of the approach presented in this work is publicly available.\t0\n",
            "0.3979\tModular Approach for Rapid Identification of RNA-Based Sensors. (ACS sensors)\tDetection of metabolites in real time and in whole cells requires effective molecular sensors. In this regard, fluorogenic light-up RNAs have recently become important tools for small-molecule detection in cells. However, the construction of light-up RNA sensors is an arduous task that requires structural knowledge of both the sensor and reporter RNA. De novo strategies for selecting sensors from RNA libraries are limited and are mostly restricted to known aptamers and riboswitches. Here, we provide a solution to this problem by developing a capture-SELEX variant that allows the obtained libraries and aptamers to be linked to fluorogenic RNAs in a modular and allosteric manner. The approach is generally applicable and allows for rapid modular allosteric assembly with green- or red-shifted fluorogenic RNAs.\t0\n",
            "0.3635\tIn-Cell Association of a Bioorthogonal Tubulin. (Biomacromolecules)\tStudies of proteins from one organism in another organism's cells have shown that such exogenous proteins stick more, pointing toward coevolution of the cytoplasm and protein surface to minimize stickiness. Here we flip this question around by asking whether exogenous proteins can assemble efficiently into their target complexes in a non-native cytoplasm. We use as our model system the assembly of BtubA and BtubB from \t0\n",
            "0.2728\tHexagonal Lattices of HIV Capsid Proteins Explored by Simulations Based on a Thermodynamically Consistent Model. (The journal of physical chemistry. B)\tHIV capsid proteins (CAs) may self-assemble into a variety of shapes under in vivo and in vitro conditions. Here, we employed simulations based on a residue-level coarse-grained (CG) model with full conformational flexibility to investigate hexagonal lattices, which are the underlying structural pattern for CA aggregations. Facilitated by enhanced sampling simulations to rigorously calculate CA dimerization and polymerization affinities, we calibrated our model to reproduce the experimentally measured affinities. Using the calibrated model, we performed unbiased simulations on several large systems consisting of 1512 CA subunits, allowing reversible binding and unbinding of the CAs in a thermodynamically consistent manner. In one simulation, a preassembled hexagonal CA sheet developed spontaneous curvatures reminiscent of those observed in experiments, and the edges of the sheet exhibited local curvatures larger than those of the interior. In other simulations starting with randomly distributed CAs at different concentrations, existing CA assemblies grew by binding free capsomeres to the edges and by merging with other assemblies. At high CA concentrations, rapid establishment of predominant aggregates was followed by much slower adjustments toward more regular hexagonal lattices, with increasing numbers of intact CA hexamers and pentamers being formed. Our approach of adapting a general CG model to specific systems by using experimental binding data represents a practical and effective strategy for simulating and elucidating intricate protein aggregations.\t0\n",
            "0.2448\tRobustness and reproducibility for AI learning in biomedical sciences: RENOIR. (Scientific reports)\tArtificial intelligence (AI) techniques are increasingly applied across various domains, favoured by the growing acquisition and public availability of large, complex datasets. Despite this trend, AI publications often suffer from lack of reproducibility and poor generalisation of findings, undermining scientific value and contributing to global research waste. To address these issues and focusing on the learning aspect of the AI field, we present RENOIR (REpeated random sampliNg fOr machIne leaRning), a modular open-source platform for robust and reproducible machine learning (ML) analysis. RENOIR adopts standardised pipelines for model training and testing, introducing elements of novelty, such as the dependence of the performance of the algorithm on the sample size. Additionally, RENOIR offers automated generation of transparent and usable reports, aiming to enhance the quality and reproducibility of AI studies. To demonstrate the versatility of our tool, we applied it to benchmark datasets from health, computer science, and STEM (Science, Technology, Engineering, and Mathematics) domains. Furthermore, we showcase RENOIR's successful application in recently published studies, where it identified classifiers for SET2D and TP53 mutation status in cancer. Finally, we present a use case where RENOIR was employed to address a significant pharmacological challenge-predicting drug efficacy. RENOIR is freely available at https://github.com/alebarberis/renoir .\t0\n",
            "0.2380\tSMEMO: Social Memory for Trajectory Forecasting. (IEEE transactions on pattern analysis and machine intelligence)\tEffective modeling of human interactions is of utmost importance when forecasting behaviors such as future trajectories. Each individual, with its motion, influences surrounding agents since everyone obeys to social non-written rules such as collision avoidance or group following. In this paper we model such interactions, which constantly evolve through time, by looking at the problem from an algorithmic point of view, i.e. as a data manipulation task. We present a neural network based on an end-to-end trainable working memory, which acts as an external storage where information about each agent can be continuously written, updated and recalled. We show that our method is capable of learning explainable cause-effect relationships between motions of different agents, obtaining state-of-the-art results on multiple trajectory forecasting datasets.\t0\n",
            "0.2300\tDeep learning based CETSA feature prediction cross multiple cell lines with latent space representation. (Scientific reports)\tMass spectrometry-coupled cellular thermal shift assay (MS-CETSA), a biophysical principle-based technique that measures the thermal stability of proteins at the proteome level inside the cell, has contributed significantly to the understanding of drug mechanisms of action and the dissection of protein interaction dynamics in different cellular states. One of the barriers to the wide applications of MS-CETSA is that MS-CETSA experiments must be performed on the specific cell lines of interest, which is typically time-consuming and costly in terms of labeling reagents and mass spectrometry time. In this study, we aim to predict CETSA features in various cell lines by introducing a computational framework called CycleDNN based on deep neural network technology. For a given set of n cell lines, CycleDNN comprises n auto-encoders. Each auto-encoder includes an encoder to convert CETSA features from one cell line into latent features in a latent space [Formula: see text]. It also features a decoder that transforms the latent features back into CETSA features for another cell line. In such a way, the proposed CycleDNN creates a cyclic prediction of CETSA features across different cell lines. The prediction loss, cycle-consistency loss, and latent space regularization loss are used to guide the model training. Experimental results on a public CETSA dataset demonstrate the effectiveness of our proposed approach. Furthermore, we confirm the validity of the predicted MS-CETSA data from our proposed CycleDNN through validation in protein-protein interaction prediction.\t0\n",
            "0.2177\tWhat to Make of Zero: Resolving the Statistical Noise from Conformational Reorganization in Alchemical Binding Free Energy Estimates with Metadynamics Sampling. (Journal of chemical theory and computation)\tWe introduce the self-relative binding free energy (self-RBFE) approach to evaluate the intrinsic statistical variance of dual-topology alchemical binding free energy estimators. The self-RBFE is the relative binding free energy between a ligand and a copy of the same ligand, and its true value is zero. Nevertheless, because the two copies of the ligand move independently, the self-RBFE value produced by a finite-length simulation fluctuates and can be used to measure the variance of the model. The results of this validation provide evidence that a significant fraction of the errors observed in benchmark studies reflect the statistical fluctuations of unconverged estimates rather than the models' accuracy. Furthermore, we find that ligand reorganization is a significant contributing factor to the statistical variance of binding free energy estimates and that metadynamics-accelerated conformational sampling of the torsional degrees of freedom of the ligand can drastically reduce the time to convergence.\t0\n",
            "0.2176\tDevelopment of Better Aptamers: Structured Library Approaches, Selection Methods, and Chemical Modifications. (Angewandte Chemie (International ed. in English))\tSystematic evolution of ligands by exponential enrichment (SELEX) has been used to discover thousands of aptamers since its development in 1990. Aptamers are short single-stranded oligonucleotides capable of binding to targets with high specificity and selectivity through structural recognition. While aptamers offer advantages over other molecular recognition elements in their ease of production, smaller size, extended shelf-life, and lower immunogenicity, they have yet to show significant success in real-world applications. By analyzing the importance of structured library designs, reviewing different SELEX methodologies, and the effects of chemical modifications, we provide a comprehensive overview on the production of aptamers for applications in drug delivery systems, therapeutics, diagnostics, and molecular imaging.\t0\n",
            "0.1771\tInferred regulons are consistent with regulator binding sequences in E. coli. (PLoS computational biology)\tThe transcriptional regulatory network (TRN) of E. coli consists of thousands of interactions between regulators and DNA sequences. Regulons are typically determined either from resource-intensive experimental measurement of functional binding sites, or inferred from analysis of high-throughput gene expression datasets. Recently, independent component analysis (ICA) of RNA-seq compendia has shown to be a powerful method for inferring bacterial regulons. However, it remains unclear to what extent regulons predicted by ICA structure have a biochemical basis in promoter sequences. Here, we address this question by developing machine learning models that predict inferred regulon structures in E. coli based on promoter sequence features. Models were constructed successfully (cross-validation AUROC > = 0.8) for 85% (40/47) of ICA-inferred E. coli regulons. We found that: 1) The presence of a high scoring regulator motif in the promoter region was sufficient to specify regulatory activity in 40% (19/47) of the regulons, 2) Additional features, such as DNA shape and extended motifs that can account for regulator multimeric binding, helped to specify regulon structure for the remaining 60% of regulons (28/47); 3) investigating regulons where initial machine learning models failed revealed new regulator-specific sequence features that improved model accuracy. Finally, we found that strong regulatory binding sequences underlie both the genes shared between ICA-inferred and experimental regulons as well as genes in the E. coli core pan-regulon of Fur. This work demonstrates that the structure of ICA-inferred regulons largely can be understood through the strength of regulator binding sites in promoter regions, reinforcing the utility of top-down inference for regulon discovery.\t0\n",
            "0.1431\tA benchmark of optimization solvers for genome-scale metabolic modeling of organisms and communities. (mSystems)\tGenome-scale metabolic modeling is a powerful framework for predicting metabolic phenotypes of any organism with an annotated genome. For two decades, this framework has been used for the rational design of microbial cell factories. In the last decade, the range of applications has exploded, and new frontiers have emerged, including the study of the gut microbiome and its health implications and the role of microbial communities in global ecosystems. However, all the critical steps in this framework, from model construction to simulation, require the use of powerful linear optimization solvers, with the choice often relying on commercial solvers for their well-known computational efficiency. In this work, I benchmark a total of six solvers (two commercial and four open source) and measure their performance to solve linear and mixed-integer linear problems of increasing complexity. Although commercial solvers are still the fastest, at least two open-source solvers show comparable performance. These results show that genome-scale metabolic modeling does not need to be hindered by commercial licensing schemes and can become a truly open science framework for solving urgent societal challenges.IMPORTANCEModeling the metabolism of organisms and communities allows for computational exploration of their metabolic capabilities and testing their response to genetic and environmental perturbations. This holds the potential to address multiple societal issues related to human health and the environment. One of the current limitations is the use of commercial optimization solvers with restrictive licenses for academic and non-academic use. This work compares the performance of several commercial and open-source solvers to solve some of the most complex problems in the field. Benchmarking results show that, although commercial solvers are indeed faster, some of the open-source options can also efficiently tackle the hardest problems, showing great promise for the development of open science applications.\t0\n",
            "0.1332\tGenetic algorithm-based semisupervised convolutional neural network for real-time monitoring of Escherichia coli fermentation of recombinant protein production using a Raman sensor. (Biotechnology and bioengineering)\tAs a non-destructive sensing technique, Raman spectroscopy is often combined with regression models for real-time detection of key components in microbial cultivation processes. However, achieving accurate model predictions often requires a large amount of offline measurement data for training, which is both time-consuming and labor-intensive. In order to overcome the limitations of traditional models that rely on large datasets and complex spectral preprocessing, in addition to the difficulty of training models with limited samples, we have explored a genetic algorithm-based semi-supervised convolutional neural network (GA-SCNN). GA-SCNN integrates unsupervised process spectral labeling, feature extraction, regression prediction, and transfer learning. Using only an extremely small number of offline samples of the target protein, this framework can accurately predict protein concentration, which represents a significant challenge for other models. The effectiveness of the framework has been validated in a system of Escherichia coli expressing recombinant ProA5M protein. By utilizing the labeling technique of this framework, the available dataset for glucose, lactate, ammonium ions, and optical density at 600 nm (OD600) has been expanded from 52 samples to 1302 samples. Furthermore, by introducing a small component of offline detection data for recombinant proteins into the OD600 model through transfer learning, a model for target protein detection has been retrained, providing a new direction for the development of associated models. Comparative analysis with traditional algorithms demonstrates that the GA-SCNN framework exhibits good adaptability when there is no complex spectral preprocessing. Cross-validation results confirm the robustness and high accuracy of the framework, with the predicted values of the model highly consistent with the offline measurement results.\t0\n",
            "0.1273\tHigh-Throughput Identification of Calcium Regulated Proteins Across Diverse Proteomes (biorxiv)\tCalcium ions play important roles in nearly every biological process, yet whole-proteome analysis of calcium effectors has been hindered by lack of high-throughput, unbiased, and quantitative methods to identify proteins-calcium engagement. To address this, we adapted protein thermostability assays in the budding yeast, human cells, and mouse mitochondria. Based on calcium-dependent thermostability, we identified 2884 putative calcium-regulated proteins across human, mouse, and yeast proteomes. These data revealed calcium engagement of novel signaling hubs and cellular processes, including metabolic enzymes and the spliceosome. Cross-species comparison of calcium-protein engagement and mutagenesis experiments identified residue-specific cation engagement, even within well-known EF-hand domains. Additionally, we found that the dienoyl-CoA reductase DECR1 binds calcium at physiologically-relevant concentrations with substrate-specific affinity, suggesting direct calcium regulation of mitochondrial fatty acid oxidation. These unbiased, proteomic analyses of calcium effectors establish a key resource to dissect cation engagement and its mechanistic effects across multiple species and diverse biological processes.\t0\n",
            "0.1226\tAccurate Determination of Reactivity Ratios for Copolymerization Reactions with Reversible Propagation Mechanisms  (chemrxiv)\tAn understanding of monomer sequence is required to predict and engineer the properties of copolymers. Typically, comonomer sequence is inferred or determined from reactivity ratios, which are measured through copolymerization experiments. The accurate determination of reactivity ratios from copolymerizations where one or both monomers undergo reversible propagation, however, has been complicated by difficulty in solving the underlying population balance equations, the presence of myriad copolymer equations in the literature derived under varying assumptions and simplifications, and lack of an easy to fit integrated model. Here, we rectify and assert the consistency between previously reported copolymer equations of disparate form, introduce a new method to explicitly solve the underlying population balance equations, and perform stochastic copolymerization simulations to evaluate the ability of these three methods to produce consistent comonomer consumption predictions and fits to simulated copolymerization data. We find that all methods produce consistent predictions given the same input parameters, which implies both accuracy and precision when modeling copolymerization, fitting experimental data, and making predictions of comonomer sequence. Considering this consistency, we make a recommendation to use numerical integration of the appropriate copolymer equation to fit real copolymerization data due to its ease of implementation and single subjective parameter for implementation. We further identify the minimum number of parameters required for accurate data fitting and suggest ways to measure other information ex situ from copolymerization to expediate accurate fitting. Finally, the practical utility of the methods developed herein is demonstrated through fitting seven distinct copolymerization datasets which span a wide range of copolymerization reactivities. \t0\n",
            "0.1208\tStatistical inference with a manifold-constrained RNA velocity model uncovers cell cycle speed modulations (biorxiv)\tAcross a range of biological processes, cells undergo coordinated changes in gene expression, resulting in transcriptome dynamics that unfold within a low-dimensional manifold. Single-cell RNA-sequencing (scRNA-seq) only measures temporal snapshots of gene expression. However, information on the underlying low-dimensional dynamics can be extracted using RNA velocity, which models unspliced and spliced RNA abundances to estimate the rate of change of gene expression. Available RNA velocity algorithms can be fragile and rely on heuristics that lack statistical control. Moreover, the estimated vector field is not dynamically consistent with the traversed gene expression manifold. Here, we develop a generative model of RNA velocity and a Bayesian inference approach that solves these problems. Our model couples velocity field and manifold estimation in a reformulated, unified framework, so as to coherently identify the parameters of an autonomous dynamical system. Focusing on the cell cycle, we implemented VeloCycle to study gene regulation dynamics on one-dimensional periodic manifolds and validated using live-imaging its ability to infer actual cell cycle periods. We benchmarked RNA velocity inference with sensitivity analyses and demonstrated one- and multiple-sample testing. We also conducted Markov chain Monte Carlo inference on the model, uncovering key relationships between gene-specific kinetics and our gene-independent velocity estimate. Finally, we applied VeloCycle to in vivo samples and in vitro genome-wide Perturb-seq, revealing regionally-defined proliferation modes in neural progenitors and the effect of gene knockdowns on cell cycle speed. Ultimately, VeloCycle expands the scRNA-seq analysis toolkit with a modular and statistically rigorous RNA velocity inference framework.\t0\n",
            "0.1181\tAn error correction strategy for image reconstruction by DNA sequencing microscopy. (Nature computational science)\tBy pairing adjacent molecules in situ and then mapping these pairs, DNA microscopy could substantially reduce the workload in spatial omics methods by directly inferring geometry from sequencing data alone. However, experimental artifacts can lead to errors in the adjacency data, which distort the spatial reconstruction. Here we describe a method to correct two such errors: spurious crosslinks formed between any two nodes, and fused nodes that are formed out of multiple molecules. We build on the principle that spatially close molecules should be connected and show that these errors violate this principle, allowing for their detection and correction. Our method corrects errors in simulated data, even in the presence of up to 20% errors, and proves to be more efficient at removing errors from experimental data than a read count filter. Integrating this method in DNA microscopy will substantially improve the accuracy of spatial reconstructions with lower data loss.\t0\n",
            "0.1161\tMetagenome profiling and containment estimation through abundance-corrected k-mer sketching with sylph (biorxiv)\tProfiling metagenomes against databases allows for the detection and quantification of microbes, even at low abundances where assembly is not possible. We introduce sylph (https://github.com/bluenote-1577/sylph), a metagenome profiler that estimates genome-to-metagenome containment average nucleotide identity (ANI) through zero-inflated Poisson k-mer statistics, enabling ANI-based taxa detection. Sylph is the most accurate method on the CAMI2 marine dataset, and compared to Kraken2 for multi-sample profiling, sylph takes 10x less CPU time and uses 30x less memory. Sylph's ANI estimates provide an orthogonal signal to abundance, enabling an ANI-based metagenome-wide association study for Parkinson's disease (PD) against 289,232 genomes while confirming known butyrate-PD associations at the strain level. Sylph takes < 1 minute and 16 GB of RAM to profile against 85,205 prokaryotic and 2,917,521 viral genomes, detecting 30x more viral sequences in the human gut compared to RefSeq. Sylph offers precise, efficient profiling with accurate containment ANI estimation for even low-coverage genomes.\t0\n",
            "0.0954\tBiophysical characterization and analysis of a mesophilic chorismate mutase from B. pumilus (biorxiv)\tChorismate mutases have extensively been used as computational benchmarking systems for enzyme catalysis, yet the roles entropy and enthalpy play in catalysis are still not fully understood. Thus, it is important to better understand these enzymes for potential research or industrial applications. Here, we report the first crystal structure and kinetic characterization of a chorismate mutase from Bacillus pumilus. This enzyme exhibits a high degree of similarity to a known mesophilic chorismate mutase from Bacillus subtilis. Using this crystal structure, we further employ EVB/MD simulations to construct Arrhenius plots, allowing us to extract thermodynamic activation parameters. Overall, this study provides new insights into the structural and functional features of the B. pumilus chorismate mutase and highlights its potential as a valuable enzyme for biocatalytic and biotechnological applications.\t0\n",
            "0.0923\tDODGE: Automated point source bacterial outbreak detection using cumulative long term genomic surveillance. (medrxiv)\tThe reliable and timely recognition of outbreaks is a key component of public health surveillance for foodborne diseases. Whole genome sequencing (WGS) offers high resolution typing of foodborne bacterial pathogens and facilitates the accurate detection of outbreaks. This detection relies on grouping WGS data into clusters at an appropriate genetic threshold, however, methods and tools for selecting and adjusting such thresholds according to the required resolution of surveillance and epidemiological context are lacking. Here we present DODGE (Dynamic Outbreak Detection for Genomic Epidemiology), an algorithm to dynamically select and compare these genetic thresholds. DODGE can analyse expanding datasets over time and clusters that are predicted to correspond to outbreaks (or investigation clusters) can be named with the established genomic nomenclature systems to facilitate integrated analysis across jurisdictions. DODGE was tested in two real-world genomic surveillance datasets of different duration, two months from Australia and nine years from the UK. In both cases only a minority of isolates were identified as investigation clusters. Two known outbreaks in the UK dataset were detected by DODGE and were recognised at an earlier timepoint than the outbreaks were reported. These findings demonstrated the potential of the DODGE approach to improve the effectiveness and timeliness of genomic surveillance for foodborne diseases and the effectiveness of the algorithm developed. Availability and implementation: DODGE is freely available at https://github.com/LanLab/dodge and can easily be installed using Conda.\t0\n",
            "0.0865\tDocking for Molecules That Bind in a Symmetric Stack with SymDOCK. (Journal of chemical information and modeling)\tDiscovering ligands for amyloid fibrils, such as those formed by the tau protein, is an area of great current interest. In recent structures, ligands bind in stacks in the tau fibrils to reflect the rotational and translational symmetry of the fibril itself; in these structures, the ligands make few interactions with the protein but interact extensively with each other. To exploit this symmetry and stacking, we developed SymDOCK, a method to dock molecules that follow the protein's symmetry. For each prospective ligand pose, we apply the symmetry operation of the fibril to generate a self-interacting and fibril-interacting stack, checking that doing so will not cause a clash between the original molecule and its image. Absent a clash, we retain that pose and add the ligand-ligand van der Waals energy to the ligand's docking score (here using DOCK3.8). We can check these geometries and energies using an implementation of ANI, a neural-network-based quantum-mechanical evaluation of the ligand stacking energies. In retrospective calculations, symmetry docking can reproduce the poses of three tau PET tracers whose structures have been determined. More convincingly, in a \t0\n",
            "0.0849\tEvaluation of Deep Learning for predicting rice traits using structural and single-nucleotide genomic variants (biorxiv)\tStructural variants (SVs) such as deletions, inversions, duplications, and Transposable El-ement (TE) Insertion Polymorphisms (TIPs) are prevalent in plant genomes and have played an important role in evolution and domestication, as they constitute a significant source of ge-nomic and phenotypic variability. Nevertheless, most methods in quantitative genetics that fo-cus on plant crop improvement, such as genomic prediction, consider Single Nucleotide Poly-morphisms (SNPs) as the only type of genetic marker. Here, we used rice to investigate wheth-er combining the structural and nucleotide genome-wide variation can improve prediction abil-ity of traits when compared to using only SNPs. Moreover, we also examine the potential ad-vantage of Deep Learning (DL) networks over Bayesian Linear models, which have been widely applied in genomic prediction. Specifically, the performance of BayesC and a Bayesian Repro-ducible Kernel Hilbert space regressions were compared to two different DL architectures, the Multilayer Perceptron, and the Convolution Neural Network. We further explore their prediction ability by using various marker input strategies and found that exploiting structural and nucleo-tide variation improves prediction ability on complex traits in rice. Also, DL models outper-formed Bayesian models in 75% of the studied cases. Finally, DL systematically improved pre-diction ability of binary traits against the Bayesian models.\t0\n",
            "0.0847\tLanguage models for quantum simulation. (Nature computational science)\tA key challenge in the effort to simulate today's quantum computing devices is the ability to learn and encode the complex correlations that occur between qubits. Emerging technologies based on language models adopted from machine learning have shown unique abilities to learn quantum states. We highlight the contributions that language models are making in the effort to build quantum computers and discuss their future role in the race to quantum advantage.\t0\n",
            "0.0824\tAminopyrimidine Derivatives as Multiflavivirus Antiviral Compounds Identified from a Consensus Virtual Screening Approach. (Journal of chemical information and modeling)\tAround three billion people are at risk of infection by the dengue virus (DENV) and potentially other flaviviruses. Worldwide outbreaks of DENV, Zika virus (ZIKV), and yellow fever virus (YFV), the lack of antiviral drugs, and limitations on vaccine usage emphasize the need for novel antiviral research. Here, we propose a consensus virtual screening approach to discover potential protease inhibitors (NS3\t0\n",
            "0.0809\tFitness models provide accurate short-term forecasts of SARS-CoV-2 variant frequency (medrxiv)\tGenomic surveillance of pathogen evolution is essential for public health response, treatment strategies, and vaccine development. In the context of SARS-COV-2, multiple models have been developed including Multinomial Logistic Regression (MLR) describing variant frequency growth as well as Fixed Growth Advantage (FGA), Growth Advantage Random Walk (GARW) and Piantham parameterizations describing variant Rt. These models provide estimates of variant fitness and can be used to forecast changes in variant frequency. We introduce a framework for evaluating real-time forecasts of variant frequencies, and apply this framework to the evolution of SARS-CoV-2 during 2022 in which multiple new viral variants emerged and rapidly spread through the population. We compare models across representative countries with different intensities of genomic surveillance. Retrospective assessment of model accuracy highlights that most models of variant frequency perform well and are able to produce reasonable forecasts. We find that the simple MLR model provides [~]0.6% median absolute error and [~]6% mean absolute error when forecasting 30 days out for countries with robust genomic surveillance. We investigate impacts of sequence quantity and quality across countries on forecast accuracy and conduct systematic downsampling to identify that 1000 sequences per week is fully sufficient for accurate short-term forecasts. We conclude that fitness models represent a useful prognostic tool for short-term evolutionary forecasting.\t0\n",
            "0.0766\tMetastable States in the Hinge-Bending Landscape of an Enzyme in an Atomistic Cytoplasm Simulation. (The journal of physical chemistry letters)\tMany enzymes undergo major conformational changes to function in cells, particularly when they bind to more than one substrate. We quantify the large-amplitude hinge-bending landscape of human phosphoglycerate kinase (PGK) in a human cytoplasm. Approximately 70 μs of all-atom simulations, upon coarse graining, reveal three metastable states of PGK with different hinge angle distributions and additional substates. The \"open\" state was more populated than the \"semi-open\" or \"closed\" states. In addition to free energies and barriers within the landscape, we characterized the average transition state passage time of ≈0.3 μs and reversible substrate and product binding. Human PGK in a dilute solution simulation shows a transition directly from the open to closed states, in agreement with previous SAXS experiments, suggesting that the cell-like model environment promotes stability of the human PGK semi-open state. Yeast PGK also sampled three metastable states within the cytoplasm model, with the closed state favored in our simulation.\t0\n",
            "0.0725\tChemPrice, a Python package for automated chemical price search (chemrxiv)\tThe open-access movement in chemistry has led to a surge in publicly accessible data repositories such as PubChem and ChemSpider, housing more than 100 million chemical compounds. These repositories are invaluable for large-scale virtual screening but lack crucial pricing information, a vital economic aspect influencing commercial decisions. Currently, these databases merely provide hyperlinks to vendor websites for pricing details, necessitating manual exploration due to the diverse structures of vendor websites, making automated data extraction impractical. To tackle this challenge, we introduce ChemPrice, a Python package that streamlines the collection of pricing data through the integration of ChemSpace, Mcule, and Molport platforms that provide up-to-date price information from over 100 suppliers. ChemPrice standardizes pricing units, offering consolidated and best-pricing information for specified compounds. Its modular design facilitates integration into automated virtual screening workflows, and a user-friendly web interface caters to those with limited coding skills, enabling easy compound list uploading and pricing data retrieval. This innovation addresses the complexity of collecting compound prices and enables integration into the virtual screening code workflows, providing a valuable resource for a wide community of users seeking comprehensive and up-to-date pricing information to enhance their decision-making processes.\t0\n",
            "0.0714\tAssessing the Catalytic Role of Native Glucagon Amyloid Fibrils (chemrxiv)\tGlucagon stands out as a pivotal peptide hormone, instrumental in controlling blood glucose levels and lipid metabolism. While the formation of glucagon amyloid fibrils has been documented, their biological functions remain enigmatic. Recently, we demonstrated experimentally that glucagon amyloid fibrils can act as catalysts in several biological reactions, including esterolysis, lipid hydrolysis, and dephosphorylation. Herein we present a multiscale quantum mechanics/molecular mechanics (QM/MM) simulation of the acylation step in the esterolysis of para-nitrophenyl acetate (p-NPA), catalyzed by native glucagon amyloid fibrils, serving as a model system to elucidate their catalytic function. This step entails a concerted mechanism, involving proton transfer from serine to histidine, followed by the nucleophilic attack of serine oxy anion on the carbonyl carbon of p-NPA. We computed the binding energy and free-energy profiles of this reaction using the PDLD/S-LRA-2000 and the empirical valence bond (EVB) methods. This included simulations if the reaction in an aqueous environment and in the fibril, enabling us to estimate the catalytic effect of the fibril. Our calculations obtained a barrier of 23.4 kcal∙mol–1 for the enzyme-catalyzed reaction, compared to the experimental value of 21.9 kcal∙mol–1 (and a calculated catalytic effect of 3.2 kcal∙mol–1 compared to  the  observed effect of 4.7 kcal∙mol–1) This close agreement together with the barrier reduction when transitioning from the amyloid to the reference solution reaction provides  supporting evidence to  the catalytic role of glucagon amyloid fibrils. Furthermore, by employing the PDLD/S-LRA-2000 approach further reinforced exclusively the enzyme's catalytic role. The results presented in this study contribute significantly to our understanding of the catalytic role of glucagon amyloid fibrils, marking, to the best of our knowledge, the first mechanistic investigation of fibrils using QM/MM methods. Therefore, our findings offer fruitful insights for future research on the mechanisms of related amyloid catalysis. \t0\n",
            "0.0667\tPolarizable AMOEBA Model for Simulating Mg (Journal of chemical information and modeling)\tMolecular mechanics (MM) simulations have the potential to provide detailed insights into the mechanisms of enzymes that utilize nucleotides as cofactors. In most cases, the activities of these enzymes also require the binding of divalent cations to catalytic sites. However, modeling divalent cations in MM simulations has been challenging. The inclusion of explicit polarization was considered promising, but despite improvements over nonpolarizable force fields and despite the inclusion of \"Nonbonded-fix (NB-fix)\" corrections, errors in interaction energies of divalent cations with proteins remain large. Importantly, the application of these models fails to reproduce the experimental structural data on Mg\t0\n",
            "0.0660\tAggregation of recount3 RNA-seq data improves the inference of consensus and tissue-specific gene co-expression networks. (biorxiv)\tBackground: Gene co-expression networks (GCNs) describe relationships among expressed genes key to maintaining cellular identity and homeostasis. However, the small sample size of typical RNA-seq experiments which is several orders of magnitude fewer than the number of genes is too low to infer GCNs reliably. recount3, a publicly available dataset comprised of 316,443 uniformly processed human RNA-seq samples, provides an opportunity to improve power for accurate network reconstruction and obtain biological insight from the resulting networks. Results: We compared alternate aggregation strategies to identify an optimal workflow for GCN inference by data aggregation and inferred three consensus networks: a universal network, a non-cancer network, and a cancer network in addition to 27 tissue context-specific networks. Central network genes from our consensus networks were enriched for evolutionarily constrained genes and ubiquitous biological pathways, whereas central context-specific network genes included tissue-specific transcription factors and factorization based on the hubs led to clustering of related tissue contexts. We discovered that annotations corresponding to context-specific networks inferred from aggregated data were enriched for trait heritability beyond known functional genomic annotations and were significantly more enriched when we aggregated over a larger number of samples. Conclusion: This study outlines best practices for network GCN inference and evaluation by data aggregation. We recommend estimating and regressing confounders in each data set before aggregation and prioritizing large sample size studies for GCN reconstruction. Increased statistical power in inferring context-specific networks enabled the derivation of variant annotations that were enriched for concordant trait heritability independent of functional genomic annotations that are context-agnostic. While we observed strictly increasing held-out log-likelihood with data aggregation, we noted diminishing marginal improvements. Future directions aimed at alternate methods for estimating confounders and integrating orthogonal information from modalities such as Hi-C and ChIP-seq can further improve GCN inference.\t0\n",
            "0.0645\tCellTracksColab--A platform for compiling, analyzing, and exploring tracking data (biorxiv)\tIn life sciences, tracking objects from movies enables researchers to quantify the behavior of single particles, organelles, bacteria, cells, and even whole animals. While numerous tools now allow automated tracking from video, a significant challenge persists in compiling, analyzing, and exploring the large datasets generated by these approaches. Here, we introduce CellTracksColab, an online platform tailored to simplify the exploration and analysis of tracking data by leveraging the free, cloud-based computational resources of Google Colab. CellTracksColab facilitates the amalgamation and analysis of results across multiple fields of view, conditions, and repeats, ensuring a holistic dataset overview. CellTracksColab also harnesses the power of high-dimensional data reduction and clustering, enabling researchers to identify distinct behavioral patterns and trends without bias. Finally, CellTracksColab also includes specialized analysis modules enabling spatial analyses (clustering, proximity to specific regions of interest). CellTracksColab is available for the broader scientific community at https://github.com/guijacquemet/CellTracksColab.\t0\n",
            "0.0609\tTrieDedup: A fast trie-based deduplication algorithm to handle ambiguous bases in high-throughput sequencing (biorxiv)\tBackground: High-throughput sequencing is a powerful tool that is extensively applied in biological studies. However, sequencers may produce low-quality bases, leading to ambiguous bases, 'N's. PCR duplicates introduced in library preparation should usually be removed in genomics studies, and several deduplication tools have been developed for this purpose. However, two identical reads may appear different due to ambiguous bases and the existing tools cannot address 'N's correctly or efficiently. Results: Here we proposed and implemented TrieDedup, which uses trie (prefix tree) data structure to compare and store sequences. TrieDedup can handle ambiguous base 'N's, and efficiently deduplicate at the level of raw sequences. We also reduced its memory usage by approximately 20% by implementing restrictedDict in Python. We benchmarked the performance of the algorithm and showed that TrieDedup can deduplicate reads up to 270-fold faster than pairwise comparison at a cost of 32-fold higher memory usage. Conclusions: TrieDedup algorithm may facilitate PCR deduplication, barcode or UMI assignment and repertoire diversity analysis of large scale high-throughput sequencing datasets with its ultra-fast algorithm that can account for ambiguous bases due to sequencing errors. Availability: TrieDedup is available at https://github.com/lolrenceH/TrieDedup\t0\n",
            "0.0585\tfastDFE: fast and flexible inference of the distribution of fitness effects (biorxiv)\tEstimating the distribution of fitness effects (DFE) of new mutations is of fundamental importance in evolutionary biology, ecology, and conservation. However, existing methods for DFE estimation suffer from limitations, such as slow computation speed and limited scalability. To address these issues, we introduce fastDFE, a Python-based software package, offering fast and flexible DFE inference from site-frequency spectrum (SFS) data. Apart from providing efficient joint inference of multiple DFEs that share parameters, it offers the feature of introducing genomic covariates that influence the DFEs, and testing their significance. To further simplify usage, fastDFE is equipped with comprehensive VCF-to-SFS parsing utilities. These include options for site filtering and stratification, as well as site-degeneracy annotation and probabilistic ancestral-allele inference. fastDFE thereby covers the entire workflow of DFE inference from the moment of acquiring a raw VCF file. Despite its Python foundation, fastDFE comprises a full R interface, including native R visualization capabilities. The package is comprehensively tested, and documented at fastdfe.readthedocs.io.\t0\n",
            "0.0582\tpSPICA Force Field Extended for Proteins and Peptides. (Journal of chemical information and modeling)\tMany coarse-grained (CG) molecular dynamics (MD) studies have been performed to investigate biological processes involving proteins and lipids. CG force fields (FFs) in these MD studies often use implicit or nonpolar water models to reduce computational costs. CG-MD using water models cannot properly describe electrostatic screening effects owing to the hydration of ionic segments and thus cannot appropriately describe molecular events involving water channels and pores through lipid membranes. To overcome this issue, we developed a protein model in the pSPICA FF, in which a polar CG water model showing the proper dielectric response was adopted. The developed CG model greatly improved the transfer free energy profiles of charged side chain analogues across the lipid membrane. Application studies on melittin-induced membrane pores and mechanosensitive channels in lipid membranes demonstrated that CG-MDs using the pSPICA FF correctly reproduced the structure and stability of the pores and channels. Furthermore, the adsorption behavior of the highly charged nona-arginine peptides on lipid membranes changed with salt concentration, indicating the pSPICA FF is also useful for simulating protein adsorption on membrane surfaces.\t0\n",
            "0.0555\tCell Type-Agnostic Optical Perturbation Screening Using Nuclear In-Situ Sequencing (NIS-Seq) (biorxiv)\tGenome-scale perturbation screening is widely used to identify disease-relevant cellular proteins serving as potential drug targets. However, most biological processes are not compatible with commonly employed perturbation screening methods, which rely on FACS- or growth-based enrichment of cells. Optical pooled screening instead uses fluorescence microscopy to determine the phenotype in single cells, and subsequently to identify individual perturbagens in the same cells. Published methods rely on cytosolic detection of endogenously expressed barcoded transcripts, which limits application to large, transcriptionally active cell types, and often relies on local clusters of clonal cells for unequivocal barcode assignment, thus precluding genome-scale screening for many biological processes. Nuclear In-Situ Sequencing (NIS-Seq) solves these shortcomings by creating bright sequencing signals directly from nuclear genomic DNA, enabling screening any nucleus-containing cell type at high density and high library complexity. We benchmark NIS-Seq by performing three genome-scale optical screens in live cells, identifying key players of inflammation-related cellular pathways.\t0\n",
            "0.0509\tA mean-field theory for predicting single polymer collapse induced by neutral crowders (biorxiv)\tMacromolecular crowding can induce the collapse of a single long polymer into a globular form due to depletion forces of entropic nature. This phenomenon has been shown to play a significant role in compacting the genome within the bacterium Escherichia coli into a well-defined region of the cell known as the \"nucleoid\". Motivated by the biological significance of this process, numerous theoretical and computational studies have searched for the primary determinants of the behavior of polymer-crowder phases. However, our understanding of this process remains incomplete and there is debate on a quantitatively unified description. In particular, different simulation studies with explicit crowders have proposed different order parameters as potential predictors for the collapse transition. In this work, we present a comprehensive analysis of published simulation data obtained from different sources. Based on the common behavior we find in this data, we develop a unified phenomenological model that we show to be predictive. Finally, to further validate the accuracy of the model, we conduct new simulations on polymers of various sizes, and investigate the role of jamming of the crowders.\t0\n",
            "0.0503\tSegment anything in medical images. (Nature communications)\tMedical image segmentation is a critical component in clinical practice, facilitating accurate diagnosis, treatment planning, and disease monitoring. However, existing methods, often tailored to specific modalities or disease types, lack generalizability across the diverse spectrum of medical image segmentation tasks. Here we present MedSAM, a foundation model designed for bridging this gap by enabling universal medical image segmentation. The model is developed on a large-scale medical image dataset with 1,570,263 image-mask pairs, covering 10 imaging modalities and over 30 cancer types. We conduct a comprehensive evaluation on 86 internal validation tasks and 60 external validation tasks, demonstrating better accuracy and robustness than modality-wise specialist models. By delivering accurate and efficient segmentation across a wide spectrum of tasks, MedSAM holds significant potential to expedite the evolution of diagnostic tools and the personalization of treatment plans.\t0\n"
          ]
        }
      ],
      "source": [
        "# Format for adding to TSV for subsequent days.\n",
        "# Manually set the value at the end of each row to one to indicate a positive example.\n",
        "for _, row in papers[papers[\"Probability\"] >= cutoff].iterrows():\n",
        "  prob, title, journal, abstract =  row[\"Probability\"], row[\"Title\"], row[\"Journal\"], row[\"Abstract\"]\n",
        "  print(f\"{prob:.4f}\\t{title} ({journal})\\t{abstract}\\t0\")"
      ]
    },
    {
      "cell_type": "code",
      "source": [],
      "metadata": {
        "id": "yQ49R9WXbWOq"
      },
      "execution_count": 13,
      "outputs": []
    }
  ],
  "metadata": {
    "colab": {
      "provenance": [],
      "gpuType": "T4"
    },
    "kernelspec": {
      "display_name": "Python 3",
      "name": "python3"
    },
    "language_info": {
      "name": "python"
    },
    "widgets": {
      "application/vnd.jupyter.widget-state+json": {
        "a4412ef094074c5d99da2d1b533f8490": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "HBoxModel",
          "model_module_version": "1.5.0",
          "state": {
            "_dom_classes": [],
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "HBoxModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/controls",
            "_view_module_version": "1.5.0",
            "_view_name": "HBoxView",
            "box_style": "",
            "children": [
              "IPY_MODEL_60fe81830a6b4caf9b21ec5622f12864",
              "IPY_MODEL_c96b2768610d4b28a76c95fba0fbfa9a",
              "IPY_MODEL_abe277a080104b1983d6fb2c155ac3a3"
            ],
            "layout": "IPY_MODEL_d0b51d7579934c169ede97fcfa52f43a"
          }
        },
        "60fe81830a6b4caf9b21ec5622f12864": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "HTMLModel",
          "model_module_version": "1.5.0",
          "state": {
            "_dom_classes": [],
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "HTMLModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/controls",
            "_view_module_version": "1.5.0",
            "_view_name": "HTMLView",
            "description": "",
            "description_tooltip": null,
            "layout": "IPY_MODEL_a305d73afe94405a8abf721f810cd864",
            "placeholder": "​",
            "style": "IPY_MODEL_ce5194d96a2942998c6415fcf052a3b3",
            "value": "tokenizer_config.json: 100%"
          }
        },
        "c96b2768610d4b28a76c95fba0fbfa9a": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "FloatProgressModel",
          "model_module_version": "1.5.0",
          "state": {
            "_dom_classes": [],
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "FloatProgressModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/controls",
            "_view_module_version": "1.5.0",
            "_view_name": "ProgressView",
            "bar_style": "success",
            "description": "",
            "description_tooltip": null,
            "layout": "IPY_MODEL_321ce7a551c74dd191f0dfade58aa7d0",
            "max": 28,
            "min": 0,
            "orientation": "horizontal",
            "style": "IPY_MODEL_38fa22afb65243228a3542a1ff53ce2d",
            "value": 28
          }
        },
        "abe277a080104b1983d6fb2c155ac3a3": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "HTMLModel",
          "model_module_version": "1.5.0",
          "state": {
            "_dom_classes": [],
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "HTMLModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/controls",
            "_view_module_version": "1.5.0",
            "_view_name": "HTMLView",
            "description": "",
            "description_tooltip": null,
            "layout": "IPY_MODEL_12068e8a0e6342e2903266b863be0b99",
            "placeholder": "​",
            "style": "IPY_MODEL_d3d6103e675a4995b9fa6ce3be58d78b",
            "value": " 28.0/28.0 [00:00&lt;00:00, 1.66kB/s]"
          }
        },
        "d0b51d7579934c169ede97fcfa52f43a": {
          "model_module": "@jupyter-widgets/base",
          "model_name": "LayoutModel",
          "model_module_version": "1.2.0",
          "state": {
            "_model_module": "@jupyter-widgets/base",
            "_model_module_version": "1.2.0",
            "_model_name": "LayoutModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "LayoutView",
            "align_content": null,
            "align_items": null,
            "align_self": null,
            "border": null,
            "bottom": null,
            "display": null,
            "flex": null,
            "flex_flow": null,
            "grid_area": null,
            "grid_auto_columns": null,
            "grid_auto_flow": null,
            "grid_auto_rows": null,
            "grid_column": null,
            "grid_gap": null,
            "grid_row": null,
            "grid_template_areas": null,
            "grid_template_columns": null,
            "grid_template_rows": null,
            "height": null,
            "justify_content": null,
            "justify_items": null,
            "left": null,
            "margin": null,
            "max_height": null,
            "max_width": null,
            "min_height": null,
            "min_width": null,
            "object_fit": null,
            "object_position": null,
            "order": null,
            "overflow": null,
            "overflow_x": null,
            "overflow_y": null,
            "padding": null,
            "right": null,
            "top": null,
            "visibility": null,
            "width": null
          }
        },
        "a305d73afe94405a8abf721f810cd864": {
          "model_module": "@jupyter-widgets/base",
          "model_name": "LayoutModel",
          "model_module_version": "1.2.0",
          "state": {
            "_model_module": "@jupyter-widgets/base",
            "_model_module_version": "1.2.0",
            "_model_name": "LayoutModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "LayoutView",
            "align_content": null,
            "align_items": null,
            "align_self": null,
            "border": null,
            "bottom": null,
            "display": null,
            "flex": null,
            "flex_flow": null,
            "grid_area": null,
            "grid_auto_columns": null,
            "grid_auto_flow": null,
            "grid_auto_rows": null,
            "grid_column": null,
            "grid_gap": null,
            "grid_row": null,
            "grid_template_areas": null,
            "grid_template_columns": null,
            "grid_template_rows": null,
            "height": null,
            "justify_content": null,
            "justify_items": null,
            "left": null,
            "margin": null,
            "max_height": null,
            "max_width": null,
            "min_height": null,
            "min_width": null,
            "object_fit": null,
            "object_position": null,
            "order": null,
            "overflow": null,
            "overflow_x": null,
            "overflow_y": null,
            "padding": null,
            "right": null,
            "top": null,
            "visibility": null,
            "width": null
          }
        },
        "ce5194d96a2942998c6415fcf052a3b3": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "DescriptionStyleModel",
          "model_module_version": "1.5.0",
          "state": {
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "DescriptionStyleModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "StyleView",
            "description_width": ""
          }
        },
        "321ce7a551c74dd191f0dfade58aa7d0": {
          "model_module": "@jupyter-widgets/base",
          "model_name": "LayoutModel",
          "model_module_version": "1.2.0",
          "state": {
            "_model_module": "@jupyter-widgets/base",
            "_model_module_version": "1.2.0",
            "_model_name": "LayoutModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "LayoutView",
            "align_content": null,
            "align_items": null,
            "align_self": null,
            "border": null,
            "bottom": null,
            "display": null,
            "flex": null,
            "flex_flow": null,
            "grid_area": null,
            "grid_auto_columns": null,
            "grid_auto_flow": null,
            "grid_auto_rows": null,
            "grid_column": null,
            "grid_gap": null,
            "grid_row": null,
            "grid_template_areas": null,
            "grid_template_columns": null,
            "grid_template_rows": null,
            "height": null,
            "justify_content": null,
            "justify_items": null,
            "left": null,
            "margin": null,
            "max_height": null,
            "max_width": null,
            "min_height": null,
            "min_width": null,
            "object_fit": null,
            "object_position": null,
            "order": null,
            "overflow": null,
            "overflow_x": null,
            "overflow_y": null,
            "padding": null,
            "right": null,
            "top": null,
            "visibility": null,
            "width": null
          }
        },
        "38fa22afb65243228a3542a1ff53ce2d": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "ProgressStyleModel",
          "model_module_version": "1.5.0",
          "state": {
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "ProgressStyleModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "StyleView",
            "bar_color": null,
            "description_width": ""
          }
        },
        "12068e8a0e6342e2903266b863be0b99": {
          "model_module": "@jupyter-widgets/base",
          "model_name": "LayoutModel",
          "model_module_version": "1.2.0",
          "state": {
            "_model_module": "@jupyter-widgets/base",
            "_model_module_version": "1.2.0",
            "_model_name": "LayoutModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "LayoutView",
            "align_content": null,
            "align_items": null,
            "align_self": null,
            "border": null,
            "bottom": null,
            "display": null,
            "flex": null,
            "flex_flow": null,
            "grid_area": null,
            "grid_auto_columns": null,
            "grid_auto_flow": null,
            "grid_auto_rows": null,
            "grid_column": null,
            "grid_gap": null,
            "grid_row": null,
            "grid_template_areas": null,
            "grid_template_columns": null,
            "grid_template_rows": null,
            "height": null,
            "justify_content": null,
            "justify_items": null,
            "left": null,
            "margin": null,
            "max_height": null,
            "max_width": null,
            "min_height": null,
            "min_width": null,
            "object_fit": null,
            "object_position": null,
            "order": null,
            "overflow": null,
            "overflow_x": null,
            "overflow_y": null,
            "padding": null,
            "right": null,
            "top": null,
            "visibility": null,
            "width": null
          }
        },
        "d3d6103e675a4995b9fa6ce3be58d78b": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "DescriptionStyleModel",
          "model_module_version": "1.5.0",
          "state": {
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "DescriptionStyleModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "StyleView",
            "description_width": ""
          }
        },
        "0f33d816e34d4edd9fd5e27369ba538b": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "HBoxModel",
          "model_module_version": "1.5.0",
          "state": {
            "_dom_classes": [],
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "HBoxModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/controls",
            "_view_module_version": "1.5.0",
            "_view_name": "HBoxView",
            "box_style": "",
            "children": [
              "IPY_MODEL_8a74492f5db84a18bbe17ebdc216c991",
              "IPY_MODEL_eab69e96b17e4849846e39b4732ccc63",
              "IPY_MODEL_287ae5ff62c242f7991d8c5f60f79d29"
            ],
            "layout": "IPY_MODEL_a9494b04005a45d39fdb4cd3a98e3f2a"
          }
        },
        "8a74492f5db84a18bbe17ebdc216c991": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "HTMLModel",
          "model_module_version": "1.5.0",
          "state": {
            "_dom_classes": [],
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "HTMLModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/controls",
            "_view_module_version": "1.5.0",
            "_view_name": "HTMLView",
            "description": "",
            "description_tooltip": null,
            "layout": "IPY_MODEL_ed12dab7b5a14c7ca67162934d19f44b",
            "placeholder": "​",
            "style": "IPY_MODEL_8f5cd37e602a4fc2a2b104a8f1913b70",
            "value": "config.json: 100%"
          }
        },
        "eab69e96b17e4849846e39b4732ccc63": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "FloatProgressModel",
          "model_module_version": "1.5.0",
          "state": {
            "_dom_classes": [],
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "FloatProgressModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/controls",
            "_view_module_version": "1.5.0",
            "_view_name": "ProgressView",
            "bar_style": "success",
            "description": "",
            "description_tooltip": null,
            "layout": "IPY_MODEL_293c56b6e70a462e86e90acec0295201",
            "max": 385,
            "min": 0,
            "orientation": "horizontal",
            "style": "IPY_MODEL_6516f3039c074d21b7eae37edd97a7d4",
            "value": 385
          }
        },
        "287ae5ff62c242f7991d8c5f60f79d29": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "HTMLModel",
          "model_module_version": "1.5.0",
          "state": {
            "_dom_classes": [],
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "HTMLModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/controls",
            "_view_module_version": "1.5.0",
            "_view_name": "HTMLView",
            "description": "",
            "description_tooltip": null,
            "layout": "IPY_MODEL_b6726896f1894f72b365049ddb5c22b7",
            "placeholder": "​",
            "style": "IPY_MODEL_484a6db5c1294faa9f3c76e2202d2cc2",
            "value": " 385/385 [00:00&lt;00:00, 22.5kB/s]"
          }
        },
        "a9494b04005a45d39fdb4cd3a98e3f2a": {
          "model_module": "@jupyter-widgets/base",
          "model_name": "LayoutModel",
          "model_module_version": "1.2.0",
          "state": {
            "_model_module": "@jupyter-widgets/base",
            "_model_module_version": "1.2.0",
            "_model_name": "LayoutModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "LayoutView",
            "align_content": null,
            "align_items": null,
            "align_self": null,
            "border": null,
            "bottom": null,
            "display": null,
            "flex": null,
            "flex_flow": null,
            "grid_area": null,
            "grid_auto_columns": null,
            "grid_auto_flow": null,
            "grid_auto_rows": null,
            "grid_column": null,
            "grid_gap": null,
            "grid_row": null,
            "grid_template_areas": null,
            "grid_template_columns": null,
            "grid_template_rows": null,
            "height": null,
            "justify_content": null,
            "justify_items": null,
            "left": null,
            "margin": null,
            "max_height": null,
            "max_width": null,
            "min_height": null,
            "min_width": null,
            "object_fit": null,
            "object_position": null,
            "order": null,
            "overflow": null,
            "overflow_x": null,
            "overflow_y": null,
            "padding": null,
            "right": null,
            "top": null,
            "visibility": null,
            "width": null
          }
        },
        "ed12dab7b5a14c7ca67162934d19f44b": {
          "model_module": "@jupyter-widgets/base",
          "model_name": "LayoutModel",
          "model_module_version": "1.2.0",
          "state": {
            "_model_module": "@jupyter-widgets/base",
            "_model_module_version": "1.2.0",
            "_model_name": "LayoutModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "LayoutView",
            "align_content": null,
            "align_items": null,
            "align_self": null,
            "border": null,
            "bottom": null,
            "display": null,
            "flex": null,
            "flex_flow": null,
            "grid_area": null,
            "grid_auto_columns": null,
            "grid_auto_flow": null,
            "grid_auto_rows": null,
            "grid_column": null,
            "grid_gap": null,
            "grid_row": null,
            "grid_template_areas": null,
            "grid_template_columns": null,
            "grid_template_rows": null,
            "height": null,
            "justify_content": null,
            "justify_items": null,
            "left": null,
            "margin": null,
            "max_height": null,
            "max_width": null,
            "min_height": null,
            "min_width": null,
            "object_fit": null,
            "object_position": null,
            "order": null,
            "overflow": null,
            "overflow_x": null,
            "overflow_y": null,
            "padding": null,
            "right": null,
            "top": null,
            "visibility": null,
            "width": null
          }
        },
        "8f5cd37e602a4fc2a2b104a8f1913b70": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "DescriptionStyleModel",
          "model_module_version": "1.5.0",
          "state": {
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "DescriptionStyleModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "StyleView",
            "description_width": ""
          }
        },
        "293c56b6e70a462e86e90acec0295201": {
          "model_module": "@jupyter-widgets/base",
          "model_name": "LayoutModel",
          "model_module_version": "1.2.0",
          "state": {
            "_model_module": "@jupyter-widgets/base",
            "_model_module_version": "1.2.0",
            "_model_name": "LayoutModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "LayoutView",
            "align_content": null,
            "align_items": null,
            "align_self": null,
            "border": null,
            "bottom": null,
            "display": null,
            "flex": null,
            "flex_flow": null,
            "grid_area": null,
            "grid_auto_columns": null,
            "grid_auto_flow": null,
            "grid_auto_rows": null,
            "grid_column": null,
            "grid_gap": null,
            "grid_row": null,
            "grid_template_areas": null,
            "grid_template_columns": null,
            "grid_template_rows": null,
            "height": null,
            "justify_content": null,
            "justify_items": null,
            "left": null,
            "margin": null,
            "max_height": null,
            "max_width": null,
            "min_height": null,
            "min_width": null,
            "object_fit": null,
            "object_position": null,
            "order": null,
            "overflow": null,
            "overflow_x": null,
            "overflow_y": null,
            "padding": null,
            "right": null,
            "top": null,
            "visibility": null,
            "width": null
          }
        },
        "6516f3039c074d21b7eae37edd97a7d4": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "ProgressStyleModel",
          "model_module_version": "1.5.0",
          "state": {
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "ProgressStyleModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "StyleView",
            "bar_color": null,
            "description_width": ""
          }
        },
        "b6726896f1894f72b365049ddb5c22b7": {
          "model_module": "@jupyter-widgets/base",
          "model_name": "LayoutModel",
          "model_module_version": "1.2.0",
          "state": {
            "_model_module": "@jupyter-widgets/base",
            "_model_module_version": "1.2.0",
            "_model_name": "LayoutModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "LayoutView",
            "align_content": null,
            "align_items": null,
            "align_self": null,
            "border": null,
            "bottom": null,
            "display": null,
            "flex": null,
            "flex_flow": null,
            "grid_area": null,
            "grid_auto_columns": null,
            "grid_auto_flow": null,
            "grid_auto_rows": null,
            "grid_column": null,
            "grid_gap": null,
            "grid_row": null,
            "grid_template_areas": null,
            "grid_template_columns": null,
            "grid_template_rows": null,
            "height": null,
            "justify_content": null,
            "justify_items": null,
            "left": null,
            "margin": null,
            "max_height": null,
            "max_width": null,
            "min_height": null,
            "min_width": null,
            "object_fit": null,
            "object_position": null,
            "order": null,
            "overflow": null,
            "overflow_x": null,
            "overflow_y": null,
            "padding": null,
            "right": null,
            "top": null,
            "visibility": null,
            "width": null
          }
        },
        "484a6db5c1294faa9f3c76e2202d2cc2": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "DescriptionStyleModel",
          "model_module_version": "1.5.0",
          "state": {
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "DescriptionStyleModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "StyleView",
            "description_width": ""
          }
        },
        "a4fb8c79d97a4385944c4530fd90935a": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "HBoxModel",
          "model_module_version": "1.5.0",
          "state": {
            "_dom_classes": [],
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "HBoxModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/controls",
            "_view_module_version": "1.5.0",
            "_view_name": "HBoxView",
            "box_style": "",
            "children": [
              "IPY_MODEL_df923834ed854d54aae0655efe2271c4",
              "IPY_MODEL_d83dab517c7241cba4f5423fc7328ad2",
              "IPY_MODEL_c300203ff26849858c7e72a39d1f2316"
            ],
            "layout": "IPY_MODEL_569a1eebc8394fc0ab53e616df96f155"
          }
        },
        "df923834ed854d54aae0655efe2271c4": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "HTMLModel",
          "model_module_version": "1.5.0",
          "state": {
            "_dom_classes": [],
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "HTMLModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/controls",
            "_view_module_version": "1.5.0",
            "_view_name": "HTMLView",
            "description": "",
            "description_tooltip": null,
            "layout": "IPY_MODEL_c9e20a1a2b71498c80b738036f471167",
            "placeholder": "​",
            "style": "IPY_MODEL_d6a3054e3e194337862c421b33124fd1",
            "value": "vocab.txt: 100%"
          }
        },
        "d83dab517c7241cba4f5423fc7328ad2": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "FloatProgressModel",
          "model_module_version": "1.5.0",
          "state": {
            "_dom_classes": [],
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "FloatProgressModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/controls",
            "_view_module_version": "1.5.0",
            "_view_name": "ProgressView",
            "bar_style": "success",
            "description": "",
            "description_tooltip": null,
            "layout": "IPY_MODEL_4b24874580dd419bb289011f5e9c7ea9",
            "max": 226150,
            "min": 0,
            "orientation": "horizontal",
            "style": "IPY_MODEL_43d348bd3cc44af283f89d92b2e99d6e",
            "value": 226150
          }
        },
        "c300203ff26849858c7e72a39d1f2316": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "HTMLModel",
          "model_module_version": "1.5.0",
          "state": {
            "_dom_classes": [],
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "HTMLModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/controls",
            "_view_module_version": "1.5.0",
            "_view_name": "HTMLView",
            "description": "",
            "description_tooltip": null,
            "layout": "IPY_MODEL_5e5387dbf7a04ada9c1508f811d597c9",
            "placeholder": "​",
            "style": "IPY_MODEL_c94247ead32d4182b03bd7bcbb35241d",
            "value": " 226k/226k [00:00&lt;00:00, 11.0MB/s]"
          }
        },
        "569a1eebc8394fc0ab53e616df96f155": {
          "model_module": "@jupyter-widgets/base",
          "model_name": "LayoutModel",
          "model_module_version": "1.2.0",
          "state": {
            "_model_module": "@jupyter-widgets/base",
            "_model_module_version": "1.2.0",
            "_model_name": "LayoutModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "LayoutView",
            "align_content": null,
            "align_items": null,
            "align_self": null,
            "border": null,
            "bottom": null,
            "display": null,
            "flex": null,
            "flex_flow": null,
            "grid_area": null,
            "grid_auto_columns": null,
            "grid_auto_flow": null,
            "grid_auto_rows": null,
            "grid_column": null,
            "grid_gap": null,
            "grid_row": null,
            "grid_template_areas": null,
            "grid_template_columns": null,
            "grid_template_rows": null,
            "height": null,
            "justify_content": null,
            "justify_items": null,
            "left": null,
            "margin": null,
            "max_height": null,
            "max_width": null,
            "min_height": null,
            "min_width": null,
            "object_fit": null,
            "object_position": null,
            "order": null,
            "overflow": null,
            "overflow_x": null,
            "overflow_y": null,
            "padding": null,
            "right": null,
            "top": null,
            "visibility": null,
            "width": null
          }
        },
        "c9e20a1a2b71498c80b738036f471167": {
          "model_module": "@jupyter-widgets/base",
          "model_name": "LayoutModel",
          "model_module_version": "1.2.0",
          "state": {
            "_model_module": "@jupyter-widgets/base",
            "_model_module_version": "1.2.0",
            "_model_name": "LayoutModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "LayoutView",
            "align_content": null,
            "align_items": null,
            "align_self": null,
            "border": null,
            "bottom": null,
            "display": null,
            "flex": null,
            "flex_flow": null,
            "grid_area": null,
            "grid_auto_columns": null,
            "grid_auto_flow": null,
            "grid_auto_rows": null,
            "grid_column": null,
            "grid_gap": null,
            "grid_row": null,
            "grid_template_areas": null,
            "grid_template_columns": null,
            "grid_template_rows": null,
            "height": null,
            "justify_content": null,
            "justify_items": null,
            "left": null,
            "margin": null,
            "max_height": null,
            "max_width": null,
            "min_height": null,
            "min_width": null,
            "object_fit": null,
            "object_position": null,
            "order": null,
            "overflow": null,
            "overflow_x": null,
            "overflow_y": null,
            "padding": null,
            "right": null,
            "top": null,
            "visibility": null,
            "width": null
          }
        },
        "d6a3054e3e194337862c421b33124fd1": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "DescriptionStyleModel",
          "model_module_version": "1.5.0",
          "state": {
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "DescriptionStyleModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "StyleView",
            "description_width": ""
          }
        },
        "4b24874580dd419bb289011f5e9c7ea9": {
          "model_module": "@jupyter-widgets/base",
          "model_name": "LayoutModel",
          "model_module_version": "1.2.0",
          "state": {
            "_model_module": "@jupyter-widgets/base",
            "_model_module_version": "1.2.0",
            "_model_name": "LayoutModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "LayoutView",
            "align_content": null,
            "align_items": null,
            "align_self": null,
            "border": null,
            "bottom": null,
            "display": null,
            "flex": null,
            "flex_flow": null,
            "grid_area": null,
            "grid_auto_columns": null,
            "grid_auto_flow": null,
            "grid_auto_rows": null,
            "grid_column": null,
            "grid_gap": null,
            "grid_row": null,
            "grid_template_areas": null,
            "grid_template_columns": null,
            "grid_template_rows": null,
            "height": null,
            "justify_content": null,
            "justify_items": null,
            "left": null,
            "margin": null,
            "max_height": null,
            "max_width": null,
            "min_height": null,
            "min_width": null,
            "object_fit": null,
            "object_position": null,
            "order": null,
            "overflow": null,
            "overflow_x": null,
            "overflow_y": null,
            "padding": null,
            "right": null,
            "top": null,
            "visibility": null,
            "width": null
          }
        },
        "43d348bd3cc44af283f89d92b2e99d6e": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "ProgressStyleModel",
          "model_module_version": "1.5.0",
          "state": {
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "ProgressStyleModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "StyleView",
            "bar_color": null,
            "description_width": ""
          }
        },
        "5e5387dbf7a04ada9c1508f811d597c9": {
          "model_module": "@jupyter-widgets/base",
          "model_name": "LayoutModel",
          "model_module_version": "1.2.0",
          "state": {
            "_model_module": "@jupyter-widgets/base",
            "_model_module_version": "1.2.0",
            "_model_name": "LayoutModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "LayoutView",
            "align_content": null,
            "align_items": null,
            "align_self": null,
            "border": null,
            "bottom": null,
            "display": null,
            "flex": null,
            "flex_flow": null,
            "grid_area": null,
            "grid_auto_columns": null,
            "grid_auto_flow": null,
            "grid_auto_rows": null,
            "grid_column": null,
            "grid_gap": null,
            "grid_row": null,
            "grid_template_areas": null,
            "grid_template_columns": null,
            "grid_template_rows": null,
            "height": null,
            "justify_content": null,
            "justify_items": null,
            "left": null,
            "margin": null,
            "max_height": null,
            "max_width": null,
            "min_height": null,
            "min_width": null,
            "object_fit": null,
            "object_position": null,
            "order": null,
            "overflow": null,
            "overflow_x": null,
            "overflow_y": null,
            "padding": null,
            "right": null,
            "top": null,
            "visibility": null,
            "width": null
          }
        },
        "c94247ead32d4182b03bd7bcbb35241d": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "DescriptionStyleModel",
          "model_module_version": "1.5.0",
          "state": {
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "DescriptionStyleModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "StyleView",
            "description_width": ""
          }
        },
        "a1f9c298cc2c4f7d8db6a014940d5433": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "HBoxModel",
          "model_module_version": "1.5.0",
          "state": {
            "_dom_classes": [],
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "HBoxModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/controls",
            "_view_module_version": "1.5.0",
            "_view_name": "HBoxView",
            "box_style": "",
            "children": [
              "IPY_MODEL_66b2bf28fb204af2b8d7c617936b10cb",
              "IPY_MODEL_4ac3ddb33f5841f28b6577256d96e31b",
              "IPY_MODEL_bc5007701a54460b8f85f947cefcb7e8"
            ],
            "layout": "IPY_MODEL_f781816a03294a63b0942c880e72e0b0"
          }
        },
        "66b2bf28fb204af2b8d7c617936b10cb": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "HTMLModel",
          "model_module_version": "1.5.0",
          "state": {
            "_dom_classes": [],
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "HTMLModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/controls",
            "_view_module_version": "1.5.0",
            "_view_name": "HTMLView",
            "description": "",
            "description_tooltip": null,
            "layout": "IPY_MODEL_7ff1d81e3b004e9c90d9cc5e5f7f0172",
            "placeholder": "​",
            "style": "IPY_MODEL_1d294647b2bc4a1685446e7e09910fae",
            "value": "pytorch_model.bin: 100%"
          }
        },
        "4ac3ddb33f5841f28b6577256d96e31b": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "FloatProgressModel",
          "model_module_version": "1.5.0",
          "state": {
            "_dom_classes": [],
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "FloatProgressModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/controls",
            "_view_module_version": "1.5.0",
            "_view_name": "ProgressView",
            "bar_style": "success",
            "description": "",
            "description_tooltip": null,
            "layout": "IPY_MODEL_8fe8fd9ae3f242a3951f39074d1fea7b",
            "max": 440474434,
            "min": 0,
            "orientation": "horizontal",
            "style": "IPY_MODEL_34971c178aca4603893c5a835d4bace0",
            "value": 440474434
          }
        },
        "bc5007701a54460b8f85f947cefcb7e8": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "HTMLModel",
          "model_module_version": "1.5.0",
          "state": {
            "_dom_classes": [],
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "HTMLModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/controls",
            "_view_module_version": "1.5.0",
            "_view_name": "HTMLView",
            "description": "",
            "description_tooltip": null,
            "layout": "IPY_MODEL_0e660b59b87e40b1bcedec32b748f81d",
            "placeholder": "​",
            "style": "IPY_MODEL_584bd5223e274a2bbb626b5932eb603a",
            "value": " 440M/440M [00:05&lt;00:00, 104MB/s]"
          }
        },
        "f781816a03294a63b0942c880e72e0b0": {
          "model_module": "@jupyter-widgets/base",
          "model_name": "LayoutModel",
          "model_module_version": "1.2.0",
          "state": {
            "_model_module": "@jupyter-widgets/base",
            "_model_module_version": "1.2.0",
            "_model_name": "LayoutModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "LayoutView",
            "align_content": null,
            "align_items": null,
            "align_self": null,
            "border": null,
            "bottom": null,
            "display": null,
            "flex": null,
            "flex_flow": null,
            "grid_area": null,
            "grid_auto_columns": null,
            "grid_auto_flow": null,
            "grid_auto_rows": null,
            "grid_column": null,
            "grid_gap": null,
            "grid_row": null,
            "grid_template_areas": null,
            "grid_template_columns": null,
            "grid_template_rows": null,
            "height": null,
            "justify_content": null,
            "justify_items": null,
            "left": null,
            "margin": null,
            "max_height": null,
            "max_width": null,
            "min_height": null,
            "min_width": null,
            "object_fit": null,
            "object_position": null,
            "order": null,
            "overflow": null,
            "overflow_x": null,
            "overflow_y": null,
            "padding": null,
            "right": null,
            "top": null,
            "visibility": null,
            "width": null
          }
        },
        "7ff1d81e3b004e9c90d9cc5e5f7f0172": {
          "model_module": "@jupyter-widgets/base",
          "model_name": "LayoutModel",
          "model_module_version": "1.2.0",
          "state": {
            "_model_module": "@jupyter-widgets/base",
            "_model_module_version": "1.2.0",
            "_model_name": "LayoutModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "LayoutView",
            "align_content": null,
            "align_items": null,
            "align_self": null,
            "border": null,
            "bottom": null,
            "display": null,
            "flex": null,
            "flex_flow": null,
            "grid_area": null,
            "grid_auto_columns": null,
            "grid_auto_flow": null,
            "grid_auto_rows": null,
            "grid_column": null,
            "grid_gap": null,
            "grid_row": null,
            "grid_template_areas": null,
            "grid_template_columns": null,
            "grid_template_rows": null,
            "height": null,
            "justify_content": null,
            "justify_items": null,
            "left": null,
            "margin": null,
            "max_height": null,
            "max_width": null,
            "min_height": null,
            "min_width": null,
            "object_fit": null,
            "object_position": null,
            "order": null,
            "overflow": null,
            "overflow_x": null,
            "overflow_y": null,
            "padding": null,
            "right": null,
            "top": null,
            "visibility": null,
            "width": null
          }
        },
        "1d294647b2bc4a1685446e7e09910fae": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "DescriptionStyleModel",
          "model_module_version": "1.5.0",
          "state": {
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "DescriptionStyleModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "StyleView",
            "description_width": ""
          }
        },
        "8fe8fd9ae3f242a3951f39074d1fea7b": {
          "model_module": "@jupyter-widgets/base",
          "model_name": "LayoutModel",
          "model_module_version": "1.2.0",
          "state": {
            "_model_module": "@jupyter-widgets/base",
            "_model_module_version": "1.2.0",
            "_model_name": "LayoutModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "LayoutView",
            "align_content": null,
            "align_items": null,
            "align_self": null,
            "border": null,
            "bottom": null,
            "display": null,
            "flex": null,
            "flex_flow": null,
            "grid_area": null,
            "grid_auto_columns": null,
            "grid_auto_flow": null,
            "grid_auto_rows": null,
            "grid_column": null,
            "grid_gap": null,
            "grid_row": null,
            "grid_template_areas": null,
            "grid_template_columns": null,
            "grid_template_rows": null,
            "height": null,
            "justify_content": null,
            "justify_items": null,
            "left": null,
            "margin": null,
            "max_height": null,
            "max_width": null,
            "min_height": null,
            "min_width": null,
            "object_fit": null,
            "object_position": null,
            "order": null,
            "overflow": null,
            "overflow_x": null,
            "overflow_y": null,
            "padding": null,
            "right": null,
            "top": null,
            "visibility": null,
            "width": null
          }
        },
        "34971c178aca4603893c5a835d4bace0": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "ProgressStyleModel",
          "model_module_version": "1.5.0",
          "state": {
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "ProgressStyleModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "StyleView",
            "bar_color": null,
            "description_width": ""
          }
        },
        "0e660b59b87e40b1bcedec32b748f81d": {
          "model_module": "@jupyter-widgets/base",
          "model_name": "LayoutModel",
          "model_module_version": "1.2.0",
          "state": {
            "_model_module": "@jupyter-widgets/base",
            "_model_module_version": "1.2.0",
            "_model_name": "LayoutModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "LayoutView",
            "align_content": null,
            "align_items": null,
            "align_self": null,
            "border": null,
            "bottom": null,
            "display": null,
            "flex": null,
            "flex_flow": null,
            "grid_area": null,
            "grid_auto_columns": null,
            "grid_auto_flow": null,
            "grid_auto_rows": null,
            "grid_column": null,
            "grid_gap": null,
            "grid_row": null,
            "grid_template_areas": null,
            "grid_template_columns": null,
            "grid_template_rows": null,
            "height": null,
            "justify_content": null,
            "justify_items": null,
            "left": null,
            "margin": null,
            "max_height": null,
            "max_width": null,
            "min_height": null,
            "min_width": null,
            "object_fit": null,
            "object_position": null,
            "order": null,
            "overflow": null,
            "overflow_x": null,
            "overflow_y": null,
            "padding": null,
            "right": null,
            "top": null,
            "visibility": null,
            "width": null
          }
        },
        "584bd5223e274a2bbb626b5932eb603a": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "DescriptionStyleModel",
          "model_module_version": "1.5.0",
          "state": {
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "DescriptionStyleModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "StyleView",
            "description_width": ""
          }
        }
      }
    },
    "accelerator": "GPU"
  },
  "nbformat": 4,
  "nbformat_minor": 0
}